
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Python library for genetic mutation visualization">
      
      
        <meta name="author" content="pyMut Development Team">
      
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.16">
    
    
      
        <title>Summary Plot - Guía Completa y Técnica - pyMut</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#summary-plot-guia-completa-y-tecnica" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="pyMut" class="md-header__button md-logo" aria-label="pyMut" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            pyMut
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Summary Plot - Guía Completa y Técnica
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/Luisruimor/pyMut" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Luisruimor/pyMut
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="pyMut" class="md-nav__button md-logo" aria-label="pyMut" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    pyMut
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Luisruimor/pyMut" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Luisruimor/pyMut
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Installation
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    API Reference
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            API Reference
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" >
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Core
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            Core
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Core/pymutation_class/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PyMutation Class
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Loading
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Data Loading
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Loading/read_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Read MAF Files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Loading/read_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Read VCF Files
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
        
          
          <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Export
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_3">
            <span class="md-nav__icon md-icon"></span>
            Data Export
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Export/to_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Export to MAF
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Export/to_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Export to VCF
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Combination/combine_data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data Combination
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
        
          
          <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Filters
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_5">
            <span class="md-nav__icon md-icon"></span>
            Filters
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/genomic_range/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Genomic Range Filters
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/pass_filter/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quality Filter (PASS)
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/chrom_sample_filter/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Chromosome & Sample Filter
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/tissue_expression_filter/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tissue Expression Filter
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_6" >
        
          
          <label class="md-nav__link" for="__nav_3_6" id="__nav_3_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Analysis Methods
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_6">
            <span class="md-nav__icon md-icon"></span>
            Analysis Methods
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/tmb_analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TMB Analysis
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/pfam_domains/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pfam Domains Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_6_3" >
        
          
          <label class="md-nav__link" for="__nav_3_6_3" id="__nav_3_6_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mutational Signatures
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_6_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_6_3">
            <span class="md-nav__icon md-icon"></span>
            Mutational Signatures
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/trinucleotideMatrix/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Matriz Trinucleotide
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/estimateSignatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Estimated Signatures
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/extract_signatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Extract Signatures
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/compare_signatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Compare Signatures
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_7" >
        
          
          <label class="md-nav__link" for="__nav_3_7" id="__nav_3_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_7">
            <span class="md-nav__icon md-icon"></span>
            Annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/known_cancer_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Known Cancer Gene Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_7_2" >
        
          
          <label class="md-nav__link" for="__nav_3_7_2" id="__nav_3_7_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    VEP Annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_7_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_7_2">
            <span class="md-nav__icon md-icon"></span>
            VEP Annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/VEP%20annotate/vep_annotation_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MAF annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/VEP%20annotate/vep_annotation_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VCF annotation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/pfam_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PFAM Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/actionable_mutation_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Actionable Mutation Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_8" >
        
          
          <label class="md-nav__link" for="__nav_3_8" id="__nav_3_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Visualization API
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_8">
            <span class="md-nav__icon md-icon"></span>
            Visualization API
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Visualization/summary_plot/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Summary Plot
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Example Usage
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Example Usage
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_1" >
        
          
          <label class="md-nav__link" for="__nav_4_1" id="__nav_4_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Input Data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_1">
            <span class="md-nav__icon md-icon"></span>
            Input Data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/input_read_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reading VCF Files with pyMut
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/input_read_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reading MAF Files with pyMut
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_2" >
        
          
          <label class="md-nav__link" for="__nav_4_2" id="__nav_4_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Output Data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_2">
            <span class="md-nav__icon md-icon"></span>
            Output Data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/vcf_to_maf_and_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VCF to MAF and VCF Conversion
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/maf_to_vcf_and_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MAF to VCF and MAF Conversion
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/combination_2maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Combining MAF Files with pyMut
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
        
          
          <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Filtering
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_4">
            <span class="md-nav__icon md-icon"></span>
            Filtering
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/filtering/example_filtering/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PyMutation Filtering Methods Example
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/filtering/tissue_expression/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tissue Expression Filter Test with TCGA LAML Data
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
        
          
          <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_5">
            <span class="md-nav__icon md-icon"></span>
            Annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/annotation/vep_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VEP Annotation Demo
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/annotation/cosmic_cancer_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    COSMIC Cancer Annotation Example
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/annotation/actionable_mutation_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    OncoKB Annotation Demo
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_6" >
        
          
          <label class="md-nav__link" for="__nav_4_6" id="__nav_4_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Analysis
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_6">
            <span class="md-nav__icon md-icon"></span>
            Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/analysis/mutation_burden/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tumor Mutational Burden (TMB) Analysis with PyMut-Bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/analysis/mutational_signatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Trinucleotide Matrix Analysis Demo
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/analysis/domains_pfam/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PFAM Annotation Module
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_7" >
        
          
          <label class="md-nav__link" for="__nav_4_7" id="__nav_4_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Visualization (beta)
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_7">
            <span class="md-nav__icon md-icon"></span>
            Visualization (beta)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/visualization/maf_to_variant_classification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MAF To Variant Classification Example
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/visualization/oncoplot_example/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Example of oncoplot
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/visualization/summary_plot_example/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Example of how to use the summary plot
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#arquitectura-del-summary-plot" class="md-nav__link">
    <span class="md-ellipsis">
      Arquitectura del Summary Plot
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Arquitectura del Summary Plot">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#estructura-interna" class="md-nav__link">
    <span class="md-ellipsis">
      Estructura Interna
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#implementacion-tecnica" class="md-nav__link">
    <span class="md-ellipsis">
      Implementación Técnica
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Implementación Técnica">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#clase-principal-pymutationsummary_plot" class="md-nav__link">
    <span class="md-ellipsis">
      Clase Principal: PyMutation.summary_plot()
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subvisualizaciones-detalladas" class="md-nav__link">
    <span class="md-ellipsis">
      Subvisualizaciones Detalladas
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Subvisualizaciones Detalladas">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-variant-classification-plot-superior-izquierda" class="md-nav__link">
    <span class="md-ellipsis">
      1. Variant Classification Plot (Superior Izquierda)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-variant-type-plot-superior-centro" class="md-nav__link">
    <span class="md-ellipsis">
      2. Variant Type Plot (Superior Centro)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-snv-class-plot-superior-derecha" class="md-nav__link">
    <span class="md-ellipsis">
      3. SNV Class Plot (Superior Derecha)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-variants-per-sample-plot-tmb-inferior-izquierda" class="md-nav__link">
    <span class="md-ellipsis">
      4. Variants per Sample Plot / TMB (Inferior Izquierda)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-variant-classification-summary-boxplot-inferior-centro" class="md-nav__link">
    <span class="md-ellipsis">
      5. Variant Classification Summary / Boxplot (Inferior Centro)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-top-mutated-genes-plot-inferior-derecha" class="md-nav__link">
    <span class="md-ellipsis">
      6. Top Mutated Genes Plot (Inferior Derecha)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sistema-de-colores-y-leyenda" class="md-nav__link">
    <span class="md-ellipsis">
      Sistema de Colores y Leyenda
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sistema de Colores y Leyenda">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mapeo-de-colores-consistente" class="md-nav__link">
    <span class="md-ellipsis">
      Mapeo de Colores Consistente
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#leyenda-unificada" class="md-nav__link">
    <span class="md-ellipsis">
      Leyenda Unificada
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#manejo-de-datos-y-preprocesamiento" class="md-nav__link">
    <span class="md-ellipsis">
      Manejo de Datos y Preprocesamiento
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Manejo de Datos y Preprocesamiento">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#deteccion-automatica-de-formato" class="md-nav__link">
    <span class="md-ellipsis">
      Detección Automática de Formato
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#procesamiento-de-datos" class="md-nav__link">
    <span class="md-ellipsis">
      Procesamiento de Datos
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Procesamiento de Datos">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-extraccion-de-variant-classification" class="md-nav__link">
    <span class="md-ellipsis">
      1. Extracción de Variant Classification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-extraccion-de-variant-types" class="md-nav__link">
    <span class="md-ellipsis">
      2. Extracción de Variant Types
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-normalizacion-de-columnas" class="md-nav__link">
    <span class="md-ellipsis">
      3. Normalización de Columnas
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optimizacion-de-rendimiento" class="md-nav__link">
    <span class="md-ellipsis">
      Optimización de Rendimiento
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Optimización de Rendimiento">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#estrategias-para-datasets-grandes" class="md-nav__link">
    <span class="md-ellipsis">
      Estrategias para Datasets Grandes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Estrategias para Datasets Grandes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-limitacion-de-muestras" class="md-nav__link">
    <span class="md-ellipsis">
      1. Limitación de Muestras
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-filtrado-previo-de-datos" class="md-nav__link">
    <span class="md-ellipsis">
      2. Filtrado Previo de Datos
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-reduccion-de-genes-mostrados" class="md-nav__link">
    <span class="md-ellipsis">
      3. Reducción de Genes Mostrados
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#benchmarks-de-rendimiento" class="md-nav__link">
    <span class="md-ellipsis">
      Benchmarks de Rendimiento
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gestion-de-calidad-de-imagen" class="md-nav__link">
    <span class="md-ellipsis">
      Gestión de Calidad de Imagen
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Gestión de Calidad de Imagen">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#configuracion-automatica-de-alta-calidad" class="md-nav__link">
    <span class="md-ellipsis">
      Configuración Automática de Alta Calidad
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#metodos-de-guardado" class="md-nav__link">
    <span class="md-ellipsis">
      Métodos de Guardado
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Métodos de Guardado">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-automatico-recomendado" class="md-nav__link">
    <span class="md-ellipsis">
      1. Automático (Recomendado)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-manual" class="md-nav__link">
    <span class="md-ellipsis">
      2. Manual
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-metodo-centralizado-de-pymut" class="md-nav__link">
    <span class="md-ellipsis">
      3. Método Centralizado de pyMut
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#formatos-recomendados-por-uso" class="md-nav__link">
    <span class="md-ellipsis">
      Formatos Recomendados por Uso
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#casos-de-uso-avanzados" class="md-nav__link">
    <span class="md-ellipsis">
      Casos de Uso Avanzados
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Casos de Uso Avanzados">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-analisis-comparativo-de-cohorts" class="md-nav__link">
    <span class="md-ellipsis">
      1. Análisis Comparativo de Cohorts
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-analisis-de-subgrupos-moleculares" class="md-nav__link">
    <span class="md-ellipsis">
      2. Análisis de Subgrupos Moleculares
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-analisis-temporal-o-longitudinal" class="md-nav__link">
    <span class="md-ellipsis">
      3. Análisis Temporal o Longitudinal
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#solucion-de-problemas-tecnicos" class="md-nav__link">
    <span class="md-ellipsis">
      Solución de Problemas Técnicos
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Solución de Problemas Técnicos">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#problemas-de-memoria" class="md-nav__link">
    <span class="md-ellipsis">
      Problemas de Memoria
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#problemas-de-visualizacion" class="md-nav__link">
    <span class="md-ellipsis">
      Problemas de Visualización
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#problemas-de-rendimiento" class="md-nav__link">
    <span class="md-ellipsis">
      Problemas de Rendimiento
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#integracion-con-pipelines-de-analisis" class="md-nav__link">
    <span class="md-ellipsis">
      Integración con Pipelines de Análisis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Integración con Pipelines de Análisis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#script-de-analisis-completo" class="md-nav__link">
    <span class="md-ellipsis">
      Script de Análisis Completo
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#integracion-con-jupyter-notebooks" class="md-nav__link">
    <span class="md-ellipsis">
      Integración con Jupyter Notebooks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extensiones-y-personalizacion" class="md-nav__link">
    <span class="md-ellipsis">
      Extensiones y Personalización
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Extensiones y Personalización">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modificacion-de-colores" class="md-nav__link">
    <span class="md-ellipsis">
      Modificación de Colores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#metricas-personalizadas" class="md-nav__link">
    <span class="md-ellipsis">
      Métricas Personalizadas
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="summary-plot-guia-completa-y-tecnica">Summary Plot - Guía Completa y Técnica<a class="headerlink" href="#summary-plot-guia-completa-y-tecnica" title="Permanent link">&para;</a></h1>
<p>Esta guía cubre en detalle todos los aspectos del Summary Plot de pyMut, desde la implementación técnica hasta casos de uso avanzados.</p>
<h2 id="arquitectura-del-summary-plot">Arquitectura del Summary Plot<a class="headerlink" href="#arquitectura-del-summary-plot" title="Permanent link">&para;</a></h2>
<h3 id="estructura-interna">Estructura Interna<a class="headerlink" href="#estructura-interna" title="Permanent link">&para;</a></h3>
<p>El Summary Plot de pyMut está construido como una composición de 6 visualizaciones independientes organizadas en una grilla 2x3:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Estructura de la grilla (2 filas x 3 columnas)</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="p">[</span><span class="n">Variant</span> <span class="n">Classification</span><span class="p">]</span> <span class="p">[</span><span class="n">Variant</span> <span class="n">Type</span><span class="p">]</span>       <span class="p">[</span><span class="n">SNV</span> <span class="n">Class</span><span class="p">]</span>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="p">[</span><span class="n">Variants</span> <span class="n">per</span> <span class="n">Sample</span><span class="p">]</span>    <span class="p">[</span><span class="n">Classification</span> <span class="n">Box</span><span class="p">]</span> <span class="p">[</span><span class="n">Top</span> <span class="n">Genes</span><span class="p">]</span>
</code></pre></div>
<h3 id="implementacion-tecnica">Implementación Técnica<a class="headerlink" href="#implementacion-tecnica" title="Permanent link">&para;</a></h3>
<h4 id="clase-principal-pymutationsummary_plot">Clase Principal: PyMutation.summary_plot()<a class="headerlink" href="#clase-principal-pymutationsummary_plot" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="k">def</span><span class="w"> </span><span class="nf">summary_plot</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> 
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>               <span class="n">figsize</span><span class="p">:</span> <span class="n">Tuple</span><span class="p">[</span><span class="nb">int</span><span class="p">,</span> <span class="nb">int</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="mi">16</span><span class="p">,</span> <span class="mi">12</span><span class="p">),</span>
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>               <span class="n">title</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;Mutation Summary&quot;</span><span class="p">,</span>
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>               <span class="n">max_samples</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="nb">int</span><span class="p">]</span> <span class="o">=</span> <span class="mi">200</span><span class="p">,</span>
<a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>               <span class="n">top_genes_count</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">10</span>
<a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>               <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">plt</span><span class="o">.</span><span class="n">Figure</span>
</code></pre></div>
<p><strong>Parámetros detallados:</strong></p>
<ul>
<li><code>figsize</code>: Control preciso del tamaño de figura</li>
<li>Default: (16, 12) - Balanceado para pantalla y papel</li>
<li>Recomendado para publicaciones: (20, 16) o (18, 14)</li>
<li>
<p>Mínimo recomendado: (12, 8)</p>
</li>
<li>
<p><code>title</code>: Título principal de la figura completa</p>
</li>
<li>Se aplica como <code>fig.suptitle()</code> con fontsize=16</li>
<li>
<p>Posicionamiento automático en la parte superior</p>
</li>
<li>
<p><code>max_samples</code>: Control de rendimiento para TMB plot</p>
</li>
<li>None: Muestra todas las muestras (puede ser lento)</li>
<li>Valor recomendado: 100-200 para datasets grandes</li>
<li>
<p>Toma las primeras N muestras según orden original</p>
</li>
<li>
<p><code>top_genes_count</code>: Número de genes en el plot inferior derecho</p>
</li>
<li>Default: 10 genes</li>
<li>Rango recomendado: 5-25 genes</li>
<li>Si hay menos genes que el especificado, muestra todos</li>
</ul>
<h2 id="subvisualizaciones-detalladas">Subvisualizaciones Detalladas<a class="headerlink" href="#subvisualizaciones-detalladas" title="Permanent link">&para;</a></h2>
<h3 id="1-variant-classification-plot-superior-izquierda">1. Variant Classification Plot (Superior Izquierda)<a class="headerlink" href="#1-variant-classification-plot-superior-izquierda" title="Permanent link">&para;</a></h3>
<p><strong>Función:</strong> <code>create_variant_classification_plot()</code></p>
<p><strong>Propósito:</strong> Muestra la distribución de tipos de mutaciones en el dataset completo.</p>
<p><strong>Implementación:</strong>
- Gráfico de barras horizontales
- Ordenamiento por frecuencia (más frecuente arriba)
- Colores consistentes con el resto del summary plot</p>
<p><strong>Interpretación biológica:</strong>
- <code>Missense_Mutation</code>: Mutaciones que cambian aminoácidos (más comunes)
- <code>Silent</code>: Mutaciones sinónimas (no cambian proteína)
- <code>Nonsense_Mutation</code>: Mutaciones que crean codones stop prematuros
- <code>Frame_Shift_Del/Ins</code>: Mutaciones que alteran el marco de lectura</p>
<h3 id="2-variant-type-plot-superior-centro">2. Variant Type Plot (Superior Centro)<a class="headerlink" href="#2-variant-type-plot-superior-centro" title="Permanent link">&para;</a></h3>
<p><strong>Función:</strong> <code>create_variant_type_plot()</code></p>
<p><strong>Propósito:</strong> Clasifica las mutaciones por mecanismo molecular.</p>
<p><strong>Categorías típicas:</strong>
- <code>SNP</code>: Single Nucleotide Polymorphisms (más comunes)
- <code>DEL</code>: Deleciones (pérdida de nucleótidos)
- <code>INS</code>: Inserciones (ganancia de nucleótidos)
- <code>DNP</code>: Dinucleotide Polymorphisms
- <code>TNP</code>: Trinucleotide Polymorphisms</p>
<h3 id="3-snv-class-plot-superior-derecha">3. SNV Class Plot (Superior Derecha)<a class="headerlink" href="#3-snv-class-plot-superior-derecha" title="Permanent link">&para;</a></h3>
<p><strong>Función:</strong> <code>create_snv_class_plot()</code></p>
<p><strong>Propósito:</strong> Analiza los patrones de cambios nucleotídicos específicos.</p>
<p><strong>Parámetros técnicos:</strong>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="n">ref_column</span><span class="o">=</span><span class="s2">&quot;REF&quot;</span>    <span class="c1"># Columna con alelo de referencia</span>
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="n">alt_column</span><span class="o">=</span><span class="s2">&quot;ALT&quot;</span>    <span class="c1"># Columna con alelo alternativo</span>
</code></pre></div></p>
<p><strong>Categorías generadas:</strong>
- Transiciones: A&gt;G, G&gt;A, C&gt;T, T&gt;C (más comunes, ~66%)
- Transversiones: A&gt;C, A&gt;T, G&gt;C, G&gt;T, C&gt;A, C&gt;G, T&gt;A, T&gt;G</p>
<p><strong>Análisis de signatures mutacionales:</strong> Patrones pueden indicar:
- C&gt;T: Deaminación espontánea de citosina
- G&gt;A: Replicación errónea
- A&gt;T: Daño oxidativo</p>
<h3 id="4-variants-per-sample-plot-tmb-inferior-izquierda">4. Variants per Sample Plot / TMB (Inferior Izquierda)<a class="headerlink" href="#4-variants-per-sample-plot-tmb-inferior-izquierda" title="Permanent link">&para;</a></h3>
<p><strong>Función:</strong> <code>create_variants_per_sample_plot()</code></p>
<p><strong>Propósito:</strong> Tumor Mutation Burden - distribución de carga mutacional por muestra.</p>
<p><strong>Características técnicas:</strong>
- Barras apiladas mostrando composición de mutaciones
- Línea roja horizontal: mediana del TMB
- Ordenamiento por TMB total (mayor a menor)
- Limitación configurable con <code>max_samples</code></p>
<p><strong>Interpretación clínica:</strong>
- TMB alto: Posible beneficio de inmunoterapia
- TMB bajo: Tumor con pocas mutaciones
- Patrones de TMB: Identificación de subgrupos moleculares</p>
<h3 id="5-variant-classification-summary-boxplot-inferior-centro">5. Variant Classification Summary / Boxplot (Inferior Centro)<a class="headerlink" href="#5-variant-classification-summary-boxplot-inferior-centro" title="Permanent link">&para;</a></h3>
<p><strong>Función:</strong> <code>create_variant_classification_summary_plot()</code></p>
<p><strong>Propósito:</strong> Distribución estadística de cada tipo de mutación entre muestras.</p>
<p><strong>Elementos del boxplot:</strong>
- Caja: Rango intercuartílico (IQR)
- Línea central: Mediana
- Whiskers: 1.5 × IQR desde Q1 y Q3
- Puntos: Outliers (muestras atípicas)</p>
<p><strong>Aplicaciones:</strong>
- Identificar tipos de mutación con alta variabilidad
- Detectar muestras con patrones mutacionales atípicos
- Comparar consistencia entre tipos de mutación</p>
<h3 id="6-top-mutated-genes-plot-inferior-derecha">6. Top Mutated Genes Plot (Inferior Derecha)<a class="headerlink" href="#6-top-mutated-genes-plot-inferior-derecha" title="Permanent link">&para;</a></h3>
<p><strong>Función:</strong> <code>create_top_mutated_genes_plot()</code></p>
<p><strong>Propósito:</strong> Identificar genes más frecuentemente alterados.</p>
<p><strong>Modos de conteo:</strong>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="n">mode</span><span class="o">=</span><span class="s2">&quot;variants&quot;</span>  <span class="c1"># Cuenta total de mutaciones por gen</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="n">mode</span><span class="o">=</span><span class="s2">&quot;samples&quot;</span>   <span class="c1"># Cuenta muestras afectadas por gen (prevalencia)</span>
</code></pre></div></p>
<p><strong>En Summary Plot:</strong> Siempre usa <code>mode="variants"</code></p>
<p><strong>Interpretación:</strong>
- Genes con muchas mutaciones: Posibles hotspots
- Genes driver: Frecuentemente mutados en cáncer
- Genes passenger: Mutaciones no funcionales</p>
<h2 id="sistema-de-colores-y-leyenda">Sistema de Colores y Leyenda<a class="headerlink" href="#sistema-de-colores-y-leyenda" title="Permanent link">&para;</a></h2>
<h3 id="mapeo-de-colores-consistente">Mapeo de Colores Consistente<a class="headerlink" href="#mapeo-de-colores-consistente" title="Permanent link">&para;</a></h3>
<p>El Summary Plot genera un mapa de colores único que se aplica a todas las subvisualizaciones:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Generación automática de colores</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="n">unique_variants</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Variant_Classification&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a><span class="n">cmap</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">colormaps</span><span class="p">[</span><span class="s1">&#39;tab20&#39;</span><span class="p">]</span>
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="n">variant_color_map</span> <span class="o">=</span> <span class="p">{</span><span class="n">variant</span><span class="p">:</span> <span class="n">cmap</span><span class="p">(</span><span class="n">i</span> <span class="o">%</span> <span class="mi">20</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">variant</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">unique_variants</span><span class="p">)}</span>
</code></pre></div>
<h3 id="leyenda-unificada">Leyenda Unificada<a class="headerlink" href="#leyenda-unificada" title="Permanent link">&para;</a></h3>
<ul>
<li>Posición: Parte inferior de la figura</li>
<li>Ordenamiento: Por frecuencia de mutaciones (descendente)</li>
<li>Elementos: Solo variantes presentes en los datos</li>
<li>Formato: Máximo 5 columnas para legibilidad</li>
</ul>
<h2 id="manejo-de-datos-y-preprocesamiento">Manejo de Datos y Preprocesamiento<a class="headerlink" href="#manejo-de-datos-y-preprocesamiento" title="Permanent link">&para;</a></h2>
<h3 id="deteccion-automatica-de-formato">Detección Automática de Formato<a class="headerlink" href="#deteccion-automatica-de-formato" title="Permanent link">&para;</a></h3>
<p>pyMut detecta automáticamente el formato de los datos:</p>
<p><strong>Formato Largo (Long Format):</strong>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>Hugo_Symbol | Variant_Classification | Tumor_Sample_Barcode | REF | ALT
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>GENE1      | Missense_Mutation      | SAMPLE_001          | A   | G
</code></pre></div></p>
<p><strong>Formato Ancho (Wide Format - TCGA):</strong>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>Hugo_Symbol | Variant_Classification | TCGA-AB-2001 | TCGA-AB-2002
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>GENE1      | Missense_Mutation      | A|G          | C|T
</code></pre></div></p>
<h3 id="procesamiento-de-datos">Procesamiento de Datos<a class="headerlink" href="#procesamiento-de-datos" title="Permanent link">&para;</a></h3>
<h4 id="1-extraccion-de-variant-classification">1. Extracción de Variant Classification<a class="headerlink" href="#1-extraccion-de-variant-classification" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1"># Si no existe la columna, extrae de FUNCOTATION</span>
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="n">processed_data</span> <span class="o">=</span> <span class="n">extract_variant_classifications</span><span class="p">(</span>
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">data</span><span class="p">,</span> 
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>    <span class="n">variant_column</span><span class="o">=</span><span class="s2">&quot;Variant_Classification&quot;</span><span class="p">,</span>
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>    <span class="n">funcotation_column</span><span class="o">=</span><span class="s2">&quot;FUNCOTATION&quot;</span>
<a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="p">)</span>
</code></pre></div>
<h4 id="2-extraccion-de-variant-types">2. Extracción de Variant Types<a class="headerlink" href="#2-extraccion-de-variant-types" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="n">processed_data</span> <span class="o">=</span> <span class="n">extract_variant_types</span><span class="p">(</span>
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>    <span class="n">processed_data</span><span class="p">,</span>
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>    <span class="n">variant_column</span><span class="o">=</span><span class="s2">&quot;Variant_Type&quot;</span><span class="p">,</span>
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>    <span class="n">funcotation_column</span><span class="o">=</span><span class="s2">&quot;FUNCOTATION&quot;</span>
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a><span class="p">)</span>
</code></pre></div>
<h4 id="3-normalizacion-de-columnas">3. Normalización de Columnas<a class="headerlink" href="#3-normalizacion-de-columnas" title="Permanent link">&para;</a></h4>
<ul>
<li>Búsqueda case-insensitive de nombres de columna</li>
<li>Mapeo automático de variantes de nombres comunes</li>
<li>Validación de existencia de datos requeridos</li>
</ul>
<h2 id="optimizacion-de-rendimiento">Optimización de Rendimiento<a class="headerlink" href="#optimizacion-de-rendimiento" title="Permanent link">&para;</a></h2>
<h3 id="estrategias-para-datasets-grandes">Estrategias para Datasets Grandes<a class="headerlink" href="#estrategias-para-datasets-grandes" title="Permanent link">&para;</a></h3>
<h4 id="1-limitacion-de-muestras">1. Limitación de Muestras<a class="headerlink" href="#1-limitacion-de-muestras" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="c1"># Limita el TMB plot para mejorar rendimiento</span>
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span><span class="n">max_samples</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
</code></pre></div>
<h4 id="2-filtrado-previo-de-datos">2. Filtrado Previo de Datos<a class="headerlink" href="#2-filtrado-previo-de-datos" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Filtrar antes de crear el objeto PyMutation</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="n">relevant_variants</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Missense_Mutation&#39;</span><span class="p">,</span> <span class="s1">&#39;Nonsense_Mutation&#39;</span><span class="p">,</span> <span class="s1">&#39;Frame_Shift_Del&#39;</span><span class="p">]</span>
<a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a><span class="n">filtered_data</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;Variant_Classification&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">relevant_variants</span><span class="p">)]</span>
<a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">filtered_data</span><span class="p">)</span>
</code></pre></div>
<h4 id="3-reduccion-de-genes-mostrados">3. Reducción de Genes Mostrados<a class="headerlink" href="#3-reduccion-de-genes-mostrados" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1"># Menos genes = procesamiento más rápido</span>
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span><span class="n">top_genes_count</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
</code></pre></div>
<h3 id="benchmarks-de-rendimiento">Benchmarks de Rendimiento<a class="headerlink" href="#benchmarks-de-rendimiento" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Tamaño Dataset</th>
<th>Muestras</th>
<th>Tiempo (seg)</th>
<th>Memoria (MB)</th>
</tr>
</thead>
<tbody>
<tr>
<td>Pequeño</td>
<td>&lt; 100</td>
<td>&lt; 5</td>
<td>&lt; 100</td>
</tr>
<tr>
<td>Mediano</td>
<td>100-500</td>
<td>5-15</td>
<td>100-300</td>
</tr>
<tr>
<td>Grande</td>
<td>500+</td>
<td>15-60</td>
<td>300-800</td>
</tr>
</tbody>
</table>
<h2 id="gestion-de-calidad-de-imagen">Gestión de Calidad de Imagen<a class="headerlink" href="#gestion-de-calidad-de-imagen" title="Permanent link">&para;</a></h2>
<h3 id="configuracion-automatica-de-alta-calidad">Configuración Automática de Alta Calidad<a class="headerlink" href="#configuracion-automatica-de-alta-calidad" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="c1"># Configurar una vez al inicio del script</span>
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a><span class="n">PyMutation</span><span class="o">.</span><span class="n">configure_high_quality_plots</span><span class="p">()</span>
<a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>
<a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a><span class="c1"># Configura automáticamente:</span>
<a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a><span class="c1"># - DPI = 300</span>
<a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a><span class="c1"># - bbox_inches = &#39;tight&#39;</span>
<a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a><span class="c1"># - facecolor = &#39;white&#39;</span>
<a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a><span class="c1"># - edgecolor = &#39;none&#39;</span>
</code></pre></div>
<h3 id="metodos-de-guardado">Métodos de Guardado<a class="headerlink" href="#metodos-de-guardado" title="Permanent link">&para;</a></h3>
<h4 id="1-automatico-recomendado">1. Automático (Recomendado)<a class="headerlink" href="#1-automatico-recomendado" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="n">PyMutation</span><span class="o">.</span><span class="n">configure_high_quality_plots</span><span class="p">()</span>
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">()</span>
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s2">&quot;summary.png&quot;</span><span class="p">)</span>  <span class="c1"># Automáticamente alta calidad</span>
</code></pre></div>
<h4 id="2-manual">2. Manual<a class="headerlink" href="#2-manual" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s2">&quot;summary.png&quot;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>
</code></pre></div>
<h4 id="3-metodo-centralizado-de-pymut">3. Método Centralizado de pyMut<a class="headerlink" href="#3-metodo-centralizado-de-pymut" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="n">py_mut</span><span class="o">.</span><span class="n">save_figure</span><span class="p">(</span><span class="n">fig</span><span class="p">,</span> <span class="s2">&quot;summary.png&quot;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>  <span class="c1"># Ultra alta calidad</span>
</code></pre></div>
<h3 id="formatos-recomendados-por-uso">Formatos Recomendados por Uso<a class="headerlink" href="#formatos-recomendados-por-uso" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Formato</th>
<th>Uso Principal</th>
<th>DPI</th>
<th>Tamaño</th>
</tr>
</thead>
<tbody>
<tr>
<td>PNG</td>
<td>Presentaciones, web</td>
<td>300</td>
<td>Medio</td>
</tr>
<tr>
<td>PDF</td>
<td>Publicaciones científicas</td>
<td>300</td>
<td>Grande</td>
</tr>
<tr>
<td>SVG</td>
<td>Edición, vectorial</td>
<td>N/A</td>
<td>Pequeño</td>
</tr>
<tr>
<td>TIFF</td>
<td>Impresión profesional</td>
<td>600</td>
<td>Muy grande</td>
</tr>
</tbody>
</table>
<h2 id="casos-de-uso-avanzados">Casos de Uso Avanzados<a class="headerlink" href="#casos-de-uso-avanzados" title="Permanent link">&para;</a></h2>
<h3 id="1-analisis-comparativo-de-cohorts">1. Análisis Comparativo de Cohorts<a class="headerlink" href="#1-analisis-comparativo-de-cohorts" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="c1"># Generar summary plots para múltiples cohorts</span>
<a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a><span class="n">cohorts</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;TCGA-LAML&#39;</span><span class="p">,</span> <span class="s1">&#39;TCGA-GBM&#39;</span><span class="p">,</span> <span class="s1">&#39;TCGA-BRCA&#39;</span><span class="p">]</span>
<a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>
<a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="k">for</span> <span class="n">cohort</span> <span class="ow">in</span> <span class="n">cohorts</span><span class="p">:</span>
<a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a>    <span class="n">data</span> <span class="o">=</span> <span class="n">load_cohort_data</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a>    <span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-16-7" name="__codelineno-16-7" href="#__codelineno-16-7"></a>
<a id="__codelineno-16-8" name="__codelineno-16-8" href="#__codelineno-16-8"></a>    <span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span>
<a id="__codelineno-16-9" name="__codelineno-16-9" href="#__codelineno-16-9"></a>        <span class="n">title</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">cohort</span><span class="si">}</span><span class="s2"> Mutation Landscape&quot;</span><span class="p">,</span>
<a id="__codelineno-16-10" name="__codelineno-16-10" href="#__codelineno-16-10"></a>        <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">18</span><span class="p">,</span> <span class="mi">14</span><span class="p">),</span>
<a id="__codelineno-16-11" name="__codelineno-16-11" href="#__codelineno-16-11"></a>        <span class="n">max_samples</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
<a id="__codelineno-16-12" name="__codelineno-16-12" href="#__codelineno-16-12"></a>        <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">20</span>
<a id="__codelineno-16-13" name="__codelineno-16-13" href="#__codelineno-16-13"></a>    <span class="p">)</span>
<a id="__codelineno-16-14" name="__codelineno-16-14" href="#__codelineno-16-14"></a>
<a id="__codelineno-16-15" name="__codelineno-16-15" href="#__codelineno-16-15"></a>    <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">cohort</span><span class="si">}</span><span class="s2">_summary.png&quot;</span><span class="p">)</span>
</code></pre></div>
<h3 id="2-analisis-de-subgrupos-moleculares">2. Análisis de Subgrupos Moleculares<a class="headerlink" href="#2-analisis-de-subgrupos-moleculares" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="c1"># Analizar subgrupos específicos</span>
<a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a><span class="n">subgroups</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;Molecular_Subtype&#39;</span><span class="p">)</span>
<a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a>
<a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="k">for</span> <span class="n">subtype</span><span class="p">,</span> <span class="n">subtype_data</span> <span class="ow">in</span> <span class="n">subgroups</span><span class="p">:</span>
<a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a>    <span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">subtype_data</span><span class="p">)</span>
<a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a>
<a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a>    <span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span>
<a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a>        <span class="n">title</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;Mutations in </span><span class="si">{</span><span class="n">subtype</span><span class="si">}</span><span class="s2"> Subtype&quot;</span><span class="p">,</span>
<a id="__codelineno-17-9" name="__codelineno-17-9" href="#__codelineno-17-9"></a>        <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">15</span>
<a id="__codelineno-17-10" name="__codelineno-17-10" href="#__codelineno-17-10"></a>    <span class="p">)</span>
<a id="__codelineno-17-11" name="__codelineno-17-11" href="#__codelineno-17-11"></a>
<a id="__codelineno-17-12" name="__codelineno-17-12" href="#__codelineno-17-12"></a>    <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;subtype_</span><span class="si">{</span><span class="n">subtype</span><span class="si">}</span><span class="s2">_summary.png&quot;</span><span class="p">)</span>
</code></pre></div>
<h3 id="3-analisis-temporal-o-longitudinal">3. Análisis Temporal o Longitudinal<a class="headerlink" href="#3-analisis-temporal-o-longitudinal" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="c1"># Comparar timepoints o tratamientos</span>
<a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a><span class="n">timepoints</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Baseline&#39;</span><span class="p">,</span> <span class="s1">&#39;Post_Treatment&#39;</span><span class="p">]</span>
<a id="__codelineno-18-3" name="__codelineno-18-3" href="#__codelineno-18-3"></a>
<a id="__codelineno-18-4" name="__codelineno-18-4" href="#__codelineno-18-4"></a><span class="k">for</span> <span class="n">tp</span> <span class="ow">in</span> <span class="n">timepoints</span><span class="p">:</span>
<a id="__codelineno-18-5" name="__codelineno-18-5" href="#__codelineno-18-5"></a>    <span class="n">tp_data</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;Timepoint&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">tp</span><span class="p">]</span>
<a id="__codelineno-18-6" name="__codelineno-18-6" href="#__codelineno-18-6"></a>    <span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">tp_data</span><span class="p">)</span>
<a id="__codelineno-18-7" name="__codelineno-18-7" href="#__codelineno-18-7"></a>
<a id="__codelineno-18-8" name="__codelineno-18-8" href="#__codelineno-18-8"></a>    <span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span>
<a id="__codelineno-18-9" name="__codelineno-18-9" href="#__codelineno-18-9"></a>        <span class="n">title</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;Mutations at </span><span class="si">{</span><span class="n">tp</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
<a id="__codelineno-18-10" name="__codelineno-18-10" href="#__codelineno-18-10"></a>        <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span> <span class="mi">12</span><span class="p">)</span>
<a id="__codelineno-18-11" name="__codelineno-18-11" href="#__codelineno-18-11"></a>    <span class="p">)</span>
<a id="__codelineno-18-12" name="__codelineno-18-12" href="#__codelineno-18-12"></a>
<a id="__codelineno-18-13" name="__codelineno-18-13" href="#__codelineno-18-13"></a>    <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;mutations_</span><span class="si">{</span><span class="n">tp</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span><span class="si">}</span><span class="s2">.png&quot;</span><span class="p">)</span>
</code></pre></div>
<h2 id="solucion-de-problemas-tecnicos">Solución de Problemas Técnicos<a class="headerlink" href="#solucion-de-problemas-tecnicos" title="Permanent link">&para;</a></h2>
<h3 id="problemas-de-memoria">Problemas de Memoria<a class="headerlink" href="#problemas-de-memoria" title="Permanent link">&para;</a></h3>
<p><strong>Síntoma:</strong> Error de memoria con datasets grandes
<strong>Solución:</strong>
<div class="highlight"><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="c1"># Reducir tamaño de datos procesados</span>
<a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span>
<a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a>    <span class="n">max_samples</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span>      <span class="c1"># Limitar muestras</span>
<a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">10</span>   <span class="c1"># Limitar genes</span>
<a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a><span class="p">)</span>
<a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a>
<a id="__codelineno-19-7" name="__codelineno-19-7" href="#__codelineno-19-7"></a><span class="c1"># O filtrar datos previamente</span>
<a id="__codelineno-19-8" name="__codelineno-19-8" href="#__codelineno-19-8"></a><span class="n">filtered_data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">sample</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="mi">1000</span><span class="p">)</span>  <span class="c1"># Muestra aleatoria</span>
</code></pre></div></p>
<h3 id="problemas-de-visualizacion">Problemas de Visualización<a class="headerlink" href="#problemas-de-visualizacion" title="Permanent link">&para;</a></h3>
<p><strong>Síntoma:</strong> Plots vacíos o errores de columnas
<strong>Diagnóstico:</strong>
<div class="highlight"><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Columnas disponibles: </span><span class="si">{</span><span class="n">data</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Clasificaciones: </span><span class="si">{</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;Variant_Classification&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Tipos de variante: </span><span class="si">{</span><span class="n">data</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s1">&#39;Variant_Type&#39;</span><span class="p">,</span><span class="w"> </span><span class="s1">&#39;No disponible&#39;</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</code></pre></div></p>
<h3 id="problemas-de-rendimiento">Problemas de Rendimiento<a class="headerlink" href="#problemas-de-rendimiento" title="Permanent link">&para;</a></h3>
<p><strong>Síntoma:</strong> Generación muy lenta
<strong>Optimizaciones:</strong>
<div class="highlight"><pre><span></span><code><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="c1"># 1. Reducir complejidad visual</span>
<a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>  <span class="c1"># Figura más pequeña</span>
<a id="__codelineno-21-3" name="__codelineno-21-3" href="#__codelineno-21-3"></a>
<a id="__codelineno-21-4" name="__codelineno-21-4" href="#__codelineno-21-4"></a><span class="c1"># 2. Procesar menos datos</span>
<a id="__codelineno-21-5" name="__codelineno-21-5" href="#__codelineno-21-5"></a><span class="n">data_subset</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">5000</span><span class="p">)</span>  <span class="c1"># Primeras 5000 mutaciones</span>
<a id="__codelineno-21-6" name="__codelineno-21-6" href="#__codelineno-21-6"></a>
<a id="__codelineno-21-7" name="__codelineno-21-7" href="#__codelineno-21-7"></a><span class="c1"># 3. Usar formato más eficiente</span>
<a id="__codelineno-21-8" name="__codelineno-21-8" href="#__codelineno-21-8"></a><span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">astype</span><span class="p">({</span><span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">:</span> <span class="s1">&#39;category&#39;</span><span class="p">})</span>  <span class="c1"># Categorías para strings</span>
</code></pre></div></p>
<h2 id="integracion-con-pipelines-de-analisis">Integración con Pipelines de Análisis<a class="headerlink" href="#integracion-con-pipelines-de-analisis" title="Permanent link">&para;</a></h2>
<h3 id="script-de-analisis-completo">Script de Análisis Completo<a class="headerlink" href="#script-de-analisis-completo" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a><span class="ch">#!/usr/bin/env python3</span>
<a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a><span class="sd">&quot;&quot;&quot;</span>
<a id="__codelineno-22-3" name="__codelineno-22-3" href="#__codelineno-22-3"></a><span class="sd">Pipeline completo de análisis mutacional con pyMut</span>
<a id="__codelineno-22-4" name="__codelineno-22-4" href="#__codelineno-22-4"></a><span class="sd">&quot;&quot;&quot;</span>
<a id="__codelineno-22-5" name="__codelineno-22-5" href="#__codelineno-22-5"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<a id="__codelineno-22-6" name="__codelineno-22-6" href="#__codelineno-22-6"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pyMut</span><span class="w"> </span><span class="kn">import</span> <span class="n">PyMutation</span>
<a id="__codelineno-22-7" name="__codelineno-22-7" href="#__codelineno-22-7"></a><span class="kn">import</span><span class="w"> </span><span class="nn">os</span>
<a id="__codelineno-22-8" name="__codelineno-22-8" href="#__codelineno-22-8"></a>
<a id="__codelineno-22-9" name="__codelineno-22-9" href="#__codelineno-22-9"></a><span class="k">def</span><span class="w"> </span><span class="nf">analyze_mutations</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span> <span class="n">output_dir</span><span class="p">):</span>
<a id="__codelineno-22-10" name="__codelineno-22-10" href="#__codelineno-22-10"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Análisis completo de mutaciones&quot;&quot;&quot;</span>
<a id="__codelineno-22-11" name="__codelineno-22-11" href="#__codelineno-22-11"></a>
<a id="__codelineno-22-12" name="__codelineno-22-12" href="#__codelineno-22-12"></a>    <span class="c1"># Configurar calidad</span>
<a id="__codelineno-22-13" name="__codelineno-22-13" href="#__codelineno-22-13"></a>    <span class="n">PyMutation</span><span class="o">.</span><span class="n">configure_high_quality_plots</span><span class="p">()</span>
<a id="__codelineno-22-14" name="__codelineno-22-14" href="#__codelineno-22-14"></a>
<a id="__codelineno-22-15" name="__codelineno-22-15" href="#__codelineno-22-15"></a>    <span class="c1"># Cargar y validar datos</span>
<a id="__codelineno-22-16" name="__codelineno-22-16" href="#__codelineno-22-16"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Cargando datos de </span><span class="si">{</span><span class="n">input_file</span><span class="si">}</span><span class="s2">...&quot;</span><span class="p">)</span>
<a id="__codelineno-22-17" name="__codelineno-22-17" href="#__codelineno-22-17"></a>    <span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<a id="__codelineno-22-18" name="__codelineno-22-18" href="#__codelineno-22-18"></a>
<a id="__codelineno-22-19" name="__codelineno-22-19" href="#__codelineno-22-19"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Dataset: </span><span class="si">{</span><span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> mutaciones, </span><span class="si">{</span><span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2"> columnas&quot;</span><span class="p">)</span>
<a id="__codelineno-22-20" name="__codelineno-22-20" href="#__codelineno-22-20"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Muestras únicas: </span><span class="si">{</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;Tumor_Sample_Barcode&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-22-21" name="__codelineno-22-21" href="#__codelineno-22-21"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Genes únicos: </span><span class="si">{</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-22-22" name="__codelineno-22-22" href="#__codelineno-22-22"></a>
<a id="__codelineno-22-23" name="__codelineno-22-23" href="#__codelineno-22-23"></a>    <span class="c1"># Crear análisis</span>
<a id="__codelineno-22-24" name="__codelineno-22-24" href="#__codelineno-22-24"></a>    <span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-22-25" name="__codelineno-22-25" href="#__codelineno-22-25"></a>
<a id="__codelineno-22-26" name="__codelineno-22-26" href="#__codelineno-22-26"></a>    <span class="c1"># Generar summary plot principal</span>
<a id="__codelineno-22-27" name="__codelineno-22-27" href="#__codelineno-22-27"></a>    <span class="n">summary_fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span>
<a id="__codelineno-22-28" name="__codelineno-22-28" href="#__codelineno-22-28"></a>        <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Comprehensive Mutation Analysis&quot;</span><span class="p">,</span>
<a id="__codelineno-22-29" name="__codelineno-22-29" href="#__codelineno-22-29"></a>        <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="mi">16</span><span class="p">),</span>
<a id="__codelineno-22-30" name="__codelineno-22-30" href="#__codelineno-22-30"></a>        <span class="n">max_samples</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
<a id="__codelineno-22-31" name="__codelineno-22-31" href="#__codelineno-22-31"></a>        <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">25</span>
<a id="__codelineno-22-32" name="__codelineno-22-32" href="#__codelineno-22-32"></a>    <span class="p">)</span>
<a id="__codelineno-22-33" name="__codelineno-22-33" href="#__codelineno-22-33"></a>
<a id="__codelineno-22-34" name="__codelineno-22-34" href="#__codelineno-22-34"></a>    <span class="c1"># Guardar en múltiples formatos</span>
<a id="__codelineno-22-35" name="__codelineno-22-35" href="#__codelineno-22-35"></a>    <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span> <span class="n">exist_ok</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<a id="__codelineno-22-36" name="__codelineno-22-36" href="#__codelineno-22-36"></a>
<a id="__codelineno-22-37" name="__codelineno-22-37" href="#__codelineno-22-37"></a>    <span class="n">summary_fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">output_dir</span><span class="si">}</span><span class="s2">/mutation_summary.png&quot;</span><span class="p">)</span>
<a id="__codelineno-22-38" name="__codelineno-22-38" href="#__codelineno-22-38"></a>    <span class="n">summary_fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">output_dir</span><span class="si">}</span><span class="s2">/mutation_summary.pdf&quot;</span><span class="p">)</span>
<a id="__codelineno-22-39" name="__codelineno-22-39" href="#__codelineno-22-39"></a>    <span class="n">summary_fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">output_dir</span><span class="si">}</span><span class="s2">/mutation_summary.svg&quot;</span><span class="p">)</span>
<a id="__codelineno-22-40" name="__codelineno-22-40" href="#__codelineno-22-40"></a>
<a id="__codelineno-22-41" name="__codelineno-22-41" href="#__codelineno-22-41"></a>    <span class="c1"># Generar plots individuales para análisis detallado</span>
<a id="__codelineno-22-42" name="__codelineno-22-42" href="#__codelineno-22-42"></a>    <span class="n">individual_plots</span> <span class="o">=</span> <span class="p">[</span>
<a id="__codelineno-22-43" name="__codelineno-22-43" href="#__codelineno-22-43"></a>        <span class="p">(</span><span class="s1">&#39;variant_classification&#39;</span><span class="p">,</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">variant_classification_plot</span><span class="p">),</span>
<a id="__codelineno-22-44" name="__codelineno-22-44" href="#__codelineno-22-44"></a>        <span class="p">(</span><span class="s1">&#39;tmb&#39;</span><span class="p">,</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">variants_per_sample_plot</span><span class="p">),</span>
<a id="__codelineno-22-45" name="__codelineno-22-45" href="#__codelineno-22-45"></a>        <span class="p">(</span><span class="s1">&#39;top_genes_variants&#39;</span><span class="p">,</span> <span class="k">lambda</span><span class="p">:</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">top_mutated_genes_plot</span><span class="p">(</span><span class="n">mode</span><span class="o">=</span><span class="s1">&#39;variants&#39;</span><span class="p">)),</span>
<a id="__codelineno-22-46" name="__codelineno-22-46" href="#__codelineno-22-46"></a>        <span class="p">(</span><span class="s1">&#39;top_genes_samples&#39;</span><span class="p">,</span> <span class="k">lambda</span><span class="p">:</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">top_mutated_genes_plot</span><span class="p">(</span><span class="n">mode</span><span class="o">=</span><span class="s1">&#39;samples&#39;</span><span class="p">))</span>
<a id="__codelineno-22-47" name="__codelineno-22-47" href="#__codelineno-22-47"></a>    <span class="p">]</span>
<a id="__codelineno-22-48" name="__codelineno-22-48" href="#__codelineno-22-48"></a>
<a id="__codelineno-22-49" name="__codelineno-22-49" href="#__codelineno-22-49"></a>    <span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">plot_func</span> <span class="ow">in</span> <span class="n">individual_plots</span><span class="p">:</span>
<a id="__codelineno-22-50" name="__codelineno-22-50" href="#__codelineno-22-50"></a>        <span class="n">fig</span> <span class="o">=</span> <span class="n">plot_func</span><span class="p">()</span>
<a id="__codelineno-22-51" name="__codelineno-22-51" href="#__codelineno-22-51"></a>        <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">output_dir</span><span class="si">}</span><span class="s2">/</span><span class="si">{</span><span class="n">name</span><span class="si">}</span><span class="s2">.png&quot;</span><span class="p">)</span>
<a id="__codelineno-22-52" name="__codelineno-22-52" href="#__codelineno-22-52"></a>
<a id="__codelineno-22-53" name="__codelineno-22-53" href="#__codelineno-22-53"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;✅ Análisis completado. Resultados en </span><span class="si">{</span><span class="n">output_dir</span><span class="si">}</span><span class="s2">/&quot;</span><span class="p">)</span>
<a id="__codelineno-22-54" name="__codelineno-22-54" href="#__codelineno-22-54"></a>
<a id="__codelineno-22-55" name="__codelineno-22-55" href="#__codelineno-22-55"></a><span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s2">&quot;__main__&quot;</span><span class="p">:</span>
<a id="__codelineno-22-56" name="__codelineno-22-56" href="#__codelineno-22-56"></a>    <span class="n">analyze_mutations</span><span class="p">(</span><span class="s2">&quot;mutations.tsv&quot;</span><span class="p">,</span> <span class="s2">&quot;analysis_results&quot;</span><span class="p">)</span>
</code></pre></div>
<h3 id="integracion-con-jupyter-notebooks">Integración con Jupyter Notebooks<a class="headerlink" href="#integracion-con-jupyter-notebooks" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="c1"># Configuración para notebooks</span>
<a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="o">%</span><span class="n">matplotlib</span> <span class="n">inline</span>
<a id="__codelineno-23-3" name="__codelineno-23-3" href="#__codelineno-23-3"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pyMut</span><span class="w"> </span><span class="kn">import</span> <span class="n">PyMutation</span>
<a id="__codelineno-23-4" name="__codelineno-23-4" href="#__codelineno-23-4"></a>
<a id="__codelineno-23-5" name="__codelineno-23-5" href="#__codelineno-23-5"></a><span class="c1"># Configurar una vez por sesión</span>
<a id="__codelineno-23-6" name="__codelineno-23-6" href="#__codelineno-23-6"></a><span class="n">PyMutation</span><span class="o">.</span><span class="n">configure_high_quality_plots</span><span class="p">()</span>
<a id="__codelineno-23-7" name="__codelineno-23-7" href="#__codelineno-23-7"></a>
<a id="__codelineno-23-8" name="__codelineno-23-8" href="#__codelineno-23-8"></a><span class="c1"># Análisis interactivo</span>
<a id="__codelineno-23-9" name="__codelineno-23-9" href="#__codelineno-23-9"></a><span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-23-10" name="__codelineno-23-10" href="#__codelineno-23-10"></a>
<a id="__codelineno-23-11" name="__codelineno-23-11" href="#__codelineno-23-11"></a><span class="c1"># Guardar para reporte</span>
<a id="__codelineno-23-12" name="__codelineno-23-12" href="#__codelineno-23-12"></a><span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s2">&quot;notebook_summary.png&quot;</span><span class="p">)</span>
</code></pre></div>
<h2 id="extensiones-y-personalizacion">Extensiones y Personalización<a class="headerlink" href="#extensiones-y-personalizacion" title="Permanent link">&para;</a></h2>
<h3 id="modificacion-de-colores">Modificación de Colores<a class="headerlink" href="#modificacion-de-colores" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="c1"># Personalizar mapa de colores</span>
<a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a><span class="n">custom_colors</span> <span class="o">=</span> <span class="p">{</span>
<a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a>    <span class="s1">&#39;Missense_Mutation&#39;</span><span class="p">:</span> <span class="s1">&#39;#FF6B6B&#39;</span><span class="p">,</span>
<a id="__codelineno-24-4" name="__codelineno-24-4" href="#__codelineno-24-4"></a>    <span class="s1">&#39;Silent&#39;</span><span class="p">:</span> <span class="s1">&#39;#4ECDC4&#39;</span><span class="p">,</span> 
<a id="__codelineno-24-5" name="__codelineno-24-5" href="#__codelineno-24-5"></a>    <span class="s1">&#39;Nonsense_Mutation&#39;</span><span class="p">:</span> <span class="s1">&#39;#45B7D1&#39;</span><span class="p">,</span>
<a id="__codelineno-24-6" name="__codelineno-24-6" href="#__codelineno-24-6"></a>    <span class="s1">&#39;Frame_Shift_Del&#39;</span><span class="p">:</span> <span class="s1">&#39;#96CEB4&#39;</span>
<a id="__codelineno-24-7" name="__codelineno-24-7" href="#__codelineno-24-7"></a><span class="p">}</span>
<a id="__codelineno-24-8" name="__codelineno-24-8" href="#__codelineno-24-8"></a>
<a id="__codelineno-24-9" name="__codelineno-24-9" href="#__codelineno-24-9"></a><span class="c1"># Aplicar en plots individuales</span>
<a id="__codelineno-24-10" name="__codelineno-24-10" href="#__codelineno-24-10"></a><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
<a id="__codelineno-24-11" name="__codelineno-24-11" href="#__codelineno-24-11"></a><span class="n">create_variant_classification_plot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">color_map</span><span class="o">=</span><span class="n">custom_colors</span><span class="p">)</span>
</code></pre></div>
<h3 id="metricas-personalizadas">Métricas Personalizadas<a class="headerlink" href="#metricas-personalizadas" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="c1"># Calcular métricas adicionales</span>
<a id="__codelineno-25-2" name="__codelineno-25-2" href="#__codelineno-25-2"></a><span class="k">def</span><span class="w"> </span><span class="nf">calculate_mutation_metrics</span><span class="p">(</span><span class="n">data</span><span class="p">):</span>
<a id="__codelineno-25-3" name="__codelineno-25-3" href="#__codelineno-25-3"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Calcula métricas personalizadas&quot;&quot;&quot;</span>
<a id="__codelineno-25-4" name="__codelineno-25-4" href="#__codelineno-25-4"></a>    <span class="n">metrics</span> <span class="o">=</span> <span class="p">{</span>
<a id="__codelineno-25-5" name="__codelineno-25-5" href="#__codelineno-25-5"></a>        <span class="s1">&#39;total_mutations&#39;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="p">),</span>
<a id="__codelineno-25-6" name="__codelineno-25-6" href="#__codelineno-25-6"></a>        <span class="s1">&#39;unique_genes&#39;</span><span class="p">:</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
<a id="__codelineno-25-7" name="__codelineno-25-7" href="#__codelineno-25-7"></a>        <span class="s1">&#39;unique_samples&#39;</span><span class="p">:</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Tumor_Sample_Barcode&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
<a id="__codelineno-25-8" name="__codelineno-25-8" href="#__codelineno-25-8"></a>        <span class="s1">&#39;avg_mutations_per_sample&#39;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">/</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Tumor_Sample_Barcode&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
<a id="__codelineno-25-9" name="__codelineno-25-9" href="#__codelineno-25-9"></a>        <span class="s1">&#39;most_mutated_gene&#39;</span><span class="p">:</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<a id="__codelineno-25-10" name="__codelineno-25-10" href="#__codelineno-25-10"></a>    <span class="p">}</span>
<a id="__codelineno-25-11" name="__codelineno-25-11" href="#__codelineno-25-11"></a>    <span class="k">return</span> <span class="n">metrics</span>
<a id="__codelineno-25-12" name="__codelineno-25-12" href="#__codelineno-25-12"></a>
<a id="__codelineno-25-13" name="__codelineno-25-13" href="#__codelineno-25-13"></a><span class="c1"># Usar con summary plot</span>
<a id="__codelineno-25-14" name="__codelineno-25-14" href="#__codelineno-25-14"></a><span class="n">metrics</span> <span class="o">=</span> <span class="n">calculate_mutation_metrics</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-25-15" name="__codelineno-25-15" href="#__codelineno-25-15"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">summary_plot</span><span class="p">(</span>
<a id="__codelineno-25-16" name="__codelineno-25-16" href="#__codelineno-25-16"></a>    <span class="n">title</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;Mutations Analysis - </span><span class="si">{</span><span class="n">metrics</span><span class="p">[</span><span class="s1">&#39;total_mutations&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2"> total mutations&quot;</span>
<a id="__codelineno-25-17" name="__codelineno-25-17" href="#__codelineno-25-17"></a><span class="p">)</span>
</code></pre></div>
<p>Esta guía completa cubre todos los aspectos técnicos y prácticos del Summary Plot, proporcionando una referencia exhaustiva para usuarios desde básicos hasta avanzados. </p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 pyMut Development Team
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.code.copy", "navigation.instant", "navigation.tracking", "search.highlight", "search.suggest"], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.50899def.min.js"></script>
      
    
  </body>
</html>