
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Python library for genetic mutation visualization">
      
      
        <meta name="author" content="pyMut Development Team">
      
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.16">
    
    
      
        <title>Guía Completa del Oncoplot - pyMut</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#guia-completa-del-oncoplot" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="pyMut" class="md-header__button md-logo" aria-label="pyMut" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            pyMut
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Guía Completa del Oncoplot
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/Luisruimor/pyMut" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Luisruimor/pyMut
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="pyMut" class="md-nav__button md-logo" aria-label="pyMut" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    pyMut
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Luisruimor/pyMut" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Luisruimor/pyMut
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Installation
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    API Reference
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            API Reference
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" >
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Core
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            Core
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Core/pymutation_class/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PyMutation Class
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Loading
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Data Loading
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Loading/read_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Read MAF Files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Loading/read_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Read VCF Files
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
        
          
          <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Export
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_3">
            <span class="md-nav__icon md-icon"></span>
            Data Export
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Export/to_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Export to MAF
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Export/to_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Export to VCF
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Data%20Combination/combine_data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data Combination
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
        
          
          <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Filters
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_5">
            <span class="md-nav__icon md-icon"></span>
            Filters
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/genomic_range/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Genomic Range Filters
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/pass_filter/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quality Filter (PASS)
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/chrom_sample_filter/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Chromosome & Sample Filter
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Filters/tissue_expression_filter/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tissue Expression Filter
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_6" >
        
          
          <label class="md-nav__link" for="__nav_3_6" id="__nav_3_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Analysis Methods
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_6">
            <span class="md-nav__icon md-icon"></span>
            Analysis Methods
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/tmb_analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TMB Analysis
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/pfam_domains/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pfam Domains Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_6_3" >
        
          
          <label class="md-nav__link" for="__nav_3_6_3" id="__nav_3_6_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mutational Signatures
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_6_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_6_3">
            <span class="md-nav__icon md-icon"></span>
            Mutational Signatures
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/trinucleotideMatrix/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Matriz Trinucleotide
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/estimateSignatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Estimated Signatures
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/extract_signatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Extract Signatures
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/Mutational%20Signatures/compare_signatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Compare Signatures
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_7" >
        
          
          <label class="md-nav__link" for="__nav_3_7" id="__nav_3_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_7">
            <span class="md-nav__icon md-icon"></span>
            Annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/known_cancer_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Known Cancer Gene Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_7_2" >
        
          
          <label class="md-nav__link" for="__nav_3_7_2" id="__nav_3_7_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    VEP Annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_7_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_7_2">
            <span class="md-nav__icon md-icon"></span>
            VEP Annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/VEP%20annotate/vep_annotation_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MAF annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/VEP%20annotate/vep_annotation_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VCF annotation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/pfam_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PFAM Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Analysis%20Methods/actionable_mutation_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Actionable Mutation Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_8" >
        
          
          <label class="md-nav__link" for="__nav_3_8" id="__nav_3_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Visualization API
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_8">
            <span class="md-nav__icon md-icon"></span>
            Visualization API
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/Visualization/summary_plot/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Summary Plot
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Example Usage
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Example Usage
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_1" >
        
          
          <label class="md-nav__link" for="__nav_4_1" id="__nav_4_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Input Data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_1">
            <span class="md-nav__icon md-icon"></span>
            Input Data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/input_read_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reading VCF Files with pyMut
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/input_read_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reading MAF Files with pyMut
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_2" >
        
          
          <label class="md-nav__link" for="__nav_4_2" id="__nav_4_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Output Data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_2">
            <span class="md-nav__icon md-icon"></span>
            Output Data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/vcf_to_maf_and_vcf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VCF to MAF and VCF Conversion
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/maf_to_vcf_and_maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MAF to VCF and MAF Conversion
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/data/combination_2maf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Combining MAF Files with pyMut
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
        
          
          <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Filtering
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_4">
            <span class="md-nav__icon md-icon"></span>
            Filtering
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/filtering/example_filtering/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PyMutation Filtering Methods Example
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/filtering/tissue_expression/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tissue Expression Filter Test with TCGA LAML Data
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
        
          
          <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_5">
            <span class="md-nav__icon md-icon"></span>
            Annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/annotation/vep_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VEP Annotation Demo
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/annotation/cosmic_cancer_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    COSMIC Cancer Annotation Example
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/annotation/actionable_mutation_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    OncoKB Annotation Demo
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_6" >
        
          
          <label class="md-nav__link" for="__nav_4_6" id="__nav_4_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Analysis
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_6">
            <span class="md-nav__icon md-icon"></span>
            Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/analysis/mutation_burden/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tumor Mutational Burden (TMB) Analysis with PyMut-Bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/analysis/mutational_signatures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Trinucleotide Matrix Analysis Demo
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/analysis/domains_pfam/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PFAM Annotation Module
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_7" >
        
          
          <label class="md-nav__link" for="__nav_4_7" id="__nav_4_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Visualization (beta)
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_7">
            <span class="md-nav__icon md-icon"></span>
            Visualization (beta)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/visualization/maf_to_variant_classification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MAF To Variant Classification Example
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/visualization/oncoplot_example/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Example of oncoplot
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/visualization/summary_plot_example/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Example of how to use the summary plot
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#tabla-de-contenidos" class="md-nav__link">
    <span class="md-ellipsis">
      Tabla de Contenidos
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduccion" class="md-nav__link">
    <span class="md-ellipsis">
      Introducción
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Introducción">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#caracteristicas-principales" class="md-nav__link">
    <span class="md-ellipsis">
      Características Principales
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#instalacion-y-configuracion" class="md-nav__link">
    <span class="md-ellipsis">
      Instalación y Configuración
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Instalación y Configuración">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#requisitos-previos" class="md-nav__link">
    <span class="md-ellipsis">
      Requisitos Previos
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#configuracion-de-alta-calidad" class="md-nav__link">
    <span class="md-ellipsis">
      Configuración de Alta Calidad
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uso-basico" class="md-nav__link">
    <span class="md-ellipsis">
      Uso Básico
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Uso Básico">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ejemplo-minimo" class="md-nav__link">
    <span class="md-ellipsis">
      Ejemplo Mínimo
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ejemplo-con-personalizacion" class="md-nav__link">
    <span class="md-ellipsis">
      Ejemplo con Personalización
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parametros-detallados" class="md-nav__link">
    <span class="md-ellipsis">
      Parámetros Detallados
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parámetros Detallados">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#parametros-del-metodo-oncoplot" class="md-nav__link">
    <span class="md-ellipsis">
      Parámetros del Método oncoplot()
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#validacion-de-parametros" class="md-nav__link">
    <span class="md-ellipsis">
      Validación de Parámetros
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#formatos-de-datos-soportados" class="md-nav__link">
    <span class="md-ellipsis">
      Formatos de Datos Soportados
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Formatos de Datos Soportados">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#estructura-de-datos-requerida" class="md-nav__link">
    <span class="md-ellipsis">
      Estructura de Datos Requerida
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Estructura de Datos Requerida">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columnas-obligatorias" class="md-nav__link">
    <span class="md-ellipsis">
      Columnas Obligatorias
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#columnas-de-muestra" class="md-nav__link">
    <span class="md-ellipsis">
      Columnas de Muestra
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#formatos-de-genotipo-soportados" class="md-nav__link">
    <span class="md-ellipsis">
      Formatos de Genotipo Soportados
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#estructura-de-ejemplo" class="md-nav__link">
    <span class="md-ellipsis">
      Estructura de Ejemplo
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ejemplos-avanzados" class="md-nav__link">
    <span class="md-ellipsis">
      Ejemplos Avanzados
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Ejemplos Avanzados">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#enfoque-en-genes-especificos" class="md-nav__link">
    <span class="md-ellipsis">
      Enfoque en Genes Específicos
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vista-panoramica" class="md-nav__link">
    <span class="md-ellipsis">
      Vista Panorámica
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exportacion-multi-formato" class="md-nav__link">
    <span class="md-ellipsis">
      Exportación Multi-formato
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analisis-interactivo" class="md-nav__link">
    <span class="md-ellipsis">
      Análisis Interactivo
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#personalizacion-visual" class="md-nav__link">
    <span class="md-ellipsis">
      Personalización Visual
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Personalización Visual">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#esquema-de-colores" class="md-nav__link">
    <span class="md-ellipsis">
      Esquema de Colores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tamanos-de-figura-recomendados" class="md-nav__link">
    <span class="md-ellipsis">
      Tamaños de Figura Recomendados
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interpretacion-del-oncoplot" class="md-nav__link">
    <span class="md-ellipsis">
      Interpretación del Oncoplot
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Interpretación del Oncoplot">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#lectura-del-grafico" class="md-nav__link">
    <span class="md-ellipsis">
      Lectura del Gráfico
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#patrones-importantes" class="md-nav__link">
    <span class="md-ellipsis">
      Patrones Importantes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analisis-de-multi_hit" class="md-nav__link">
    <span class="md-ellipsis">
      Análisis de Multi_Hit
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#solucion-de-problemas" class="md-nav__link">
    <span class="md-ellipsis">
      Solución de Problemas
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Solución de Problemas">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#errores-comunes" class="md-nav__link">
    <span class="md-ellipsis">
      Errores Comunes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Errores Comunes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-columnas-faltantes" class="md-nav__link">
    <span class="md-ellipsis">
      1. Columnas Faltantes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-no-se-detectan-muestras" class="md-nav__link">
    <span class="md-ellipsis">
      2. No se Detectan Muestras
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-no-hay-datos-de-mutacion" class="md-nav__link">
    <span class="md-ellipsis">
      3. No Hay Datos de Mutación
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-parametros-invalidos" class="md-nav__link">
    <span class="md-ellipsis">
      4. Parámetros Inválidos
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#debugging-paso-a-paso" class="md-nav__link">
    <span class="md-ellipsis">
      Debugging Paso a Paso
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#problemas-de-rendimiento" class="md-nav__link">
    <span class="md-ellipsis">
      Problemas de Rendimiento
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Problemas de Rendimiento">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#datasets-grandes" class="md-nav__link">
    <span class="md-ellipsis">
      Datasets Grandes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#memoria-insuficiente" class="md-nav__link">
    <span class="md-ellipsis">
      Memoria Insuficiente
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optimizacion-de-rendimiento" class="md-nav__link">
    <span class="md-ellipsis">
      Optimización de Rendimiento
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Optimización de Rendimiento">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mejores-practicas" class="md-nav__link">
    <span class="md-ellipsis">
      Mejores Prácticas
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parametros-de-rendimiento" class="md-nav__link">
    <span class="md-ellipsis">
      Parámetros de Rendimiento
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monitoreo-de-recursos" class="md-nav__link">
    <span class="md-ellipsis">
      Monitoreo de Recursos
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mas-recursos" class="md-nav__link">
    <span class="md-ellipsis">
      Más Recursos
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="guia-completa-del-oncoplot">Guía Completa del Oncoplot<a class="headerlink" href="#guia-completa-del-oncoplot" title="Permanent link">&para;</a></h1>
<p>El <strong>oncoplot</strong> (también conocido como waterfall plot en algunos contextos) es una visualización fundamental en genómica del cáncer que muestra los patrones de mutación a través de muestras y genes en formato heatmap. Esta guía cubre todo lo que necesitas saber para crear y personalizar oncoplots usando pyMut.</p>
<h2 id="tabla-de-contenidos">Tabla de Contenidos<a class="headerlink" href="#tabla-de-contenidos" title="Permanent link">&para;</a></h2>
<ol>
<li><a href="#introducción">Introducción</a></li>
<li><a href="#instalación-y-configuración">Instalación y Configuración</a></li>
<li><a href="#uso-básico">Uso Básico</a></li>
<li><a href="#parámetros-detallados">Parámetros Detallados</a></li>
<li><a href="#formatos-de-datos-soportados">Formatos de Datos Soportados</a></li>
<li><a href="#ejemplos-avanzados">Ejemplos Avanzados</a></li>
<li><a href="#personalización-visual">Personalización Visual</a></li>
<li><a href="#interpretación-del-oncoplot">Interpretación del Oncoplot</a></li>
<li><a href="#solución-de-problemas">Solución de Problemas</a></li>
<li><a href="#optimización-de-rendimiento">Optimización de Rendimiento</a></li>
</ol>
<h2 id="introduccion">Introducción<a class="headerlink" href="#introduccion" title="Permanent link">&para;</a></h2>
<p>El oncoplot es una herramienta esencial para visualizar paisajes mutacionales en datos genómicos. Proporciona una vista panorámica de:</p>
<ul>
<li><strong>Genes más frecuentemente mutados</strong></li>
<li><strong>Patrones de co-ocurrencia de mutaciones</strong></li>
<li><strong>Carga mutacional por muestra</strong></li>
<li><strong>Tipos de mutaciones por color</strong></li>
<li><strong>Distribución de mutaciones across samples</strong></li>
</ul>
<h3 id="caracteristicas-principales">Características Principales<a class="headerlink" href="#caracteristicas-principales" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Detección automática</strong> de columnas de muestra (TCGA y formato .GT)</li>
<li><strong>Soporte multiformato</strong> para genotipos (A|G, A/G, etc.)</li>
<li><strong>Detección de Multi_Hit</strong> para muestras con múltiples mutaciones</li>
<li><strong>Esquemas de colores estándar</strong> para tipos de mutación</li>
<li><strong>Ordenamiento inteligente</strong> por frecuencia de mutación</li>
<li><strong>Alta calidad de exportación</strong> (PNG, PDF, SVG)</li>
</ul>
<h2 id="instalacion-y-configuracion">Instalación y Configuración<a class="headerlink" href="#instalacion-y-configuracion" title="Permanent link">&para;</a></h2>
<h3 id="requisitos-previos">Requisitos Previos<a class="headerlink" href="#requisitos-previos" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>pip<span class="w"> </span>install<span class="w"> </span>pandas<span class="w"> </span>matplotlib<span class="w"> </span>seaborn<span class="w"> </span>numpy
</code></pre></div>
<h3 id="configuracion-de-alta-calidad">Configuración de Alta Calidad<a class="headerlink" href="#configuracion-de-alta-calidad" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pyMut</span><span class="w"> </span><span class="kn">import</span> <span class="n">PyMutation</span>
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a><span class="c1"># Configurar alta calidad automáticamente (recomendado)</span>
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a><span class="n">PyMutation</span><span class="o">.</span><span class="n">configure_high_quality_plots</span><span class="p">()</span>
</code></pre></div>
<p>Esta configuración aplica automáticamente:
- DPI = 300 para imágenes nítidas
- bbox_inches = 'tight' para layouts optimizados
- Configuraciones de fuente mejoradas</p>
<h2 id="uso-basico">Uso Básico<a class="headerlink" href="#uso-basico" title="Permanent link">&para;</a></h2>
<h3 id="ejemplo-minimo">Ejemplo Mínimo<a class="headerlink" href="#ejemplo-minimo" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pyMut</span><span class="w"> </span><span class="kn">import</span> <span class="n">PyMutation</span>
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
<a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a><span class="c1"># Cargar datos</span>
<a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a><span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;mutations.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>
<a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="c1"># Crear objeto PyMutation</span>
<a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a><span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>
<a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a><span class="c1"># Generar oncoplot básico</span>
<a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">()</span>
<a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a>
<a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a><span class="c1"># Guardar</span>
<a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a><span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;oncoplot.png&#39;</span><span class="p">)</span>
</code></pre></div>
<h3 id="ejemplo-con-personalizacion">Ejemplo con Personalización<a class="headerlink" href="#ejemplo-con-personalizacion" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Oncoplot personalizado</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Paisaje Mutacional - TCGA LAML&quot;</span><span class="p">,</span>
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>    <span class="n">max_samples</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span>
<a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span> <span class="mi">10</span><span class="p">)</span>
<a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="p">)</span>
</code></pre></div>
<h2 id="parametros-detallados">Parámetros Detallados<a class="headerlink" href="#parametros-detallados" title="Permanent link">&para;</a></h2>
<h3 id="parametros-del-metodo-oncoplot">Parámetros del Método <code>oncoplot()</code><a class="headerlink" href="#parametros-del-metodo-oncoplot" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parámetro</th>
<th>Tipo</th>
<th>Valor por Defecto</th>
<th>Descripción</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>figsize</code></td>
<td><code>Tuple[int, int]</code></td>
<td><code>(16, 10)</code></td>
<td>Tamaño de figura (ancho, alto) en pulgadas</td>
</tr>
<tr>
<td><code>title</code></td>
<td><code>str</code></td>
<td><code>"Oncoplot"</code></td>
<td>Título del gráfico</td>
</tr>
<tr>
<td><code>gene_column</code></td>
<td><code>str</code></td>
<td><code>"Hugo_Symbol"</code></td>
<td>Columna con símbolos de genes</td>
</tr>
<tr>
<td><code>variant_column</code></td>
<td><code>str</code></td>
<td><code>"Variant_Classification"</code></td>
<td>Columna con tipos de mutación</td>
</tr>
<tr>
<td><code>ref_column</code></td>
<td><code>str</code></td>
<td><code>"REF"</code></td>
<td>Columna con alelos de referencia</td>
</tr>
<tr>
<td><code>alt_column</code></td>
<td><code>str</code></td>
<td><code>"ALT"</code></td>
<td>Columna con alelos alternativos</td>
</tr>
<tr>
<td><code>top_genes_count</code></td>
<td><code>int</code></td>
<td><code>10</code></td>
<td>Número de genes más mutados a mostrar</td>
</tr>
<tr>
<td><code>max_samples</code></td>
<td><code>int</code></td>
<td><code>180</code></td>
<td>Número máximo de muestras a mostrar</td>
</tr>
</tbody>
</table>
<h3 id="validacion-de-parametros">Validación de Parámetros<a class="headerlink" href="#validacion-de-parametros" title="Permanent link">&para;</a></h3>
<p>El método incluye validación robusta:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># ✓ Válido</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span><span class="n">top_genes_count</span><span class="o">=</span><span class="mi">15</span><span class="p">,</span> <span class="n">max_samples</span><span class="o">=</span><span class="mi">50</span><span class="p">)</span>
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># ✗ Inválido - lanzará ValueError</span>
<a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a><span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span><span class="n">top_genes_count</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>         <span class="c1"># Debe ser &gt; 0</span>
<a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a><span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span><span class="n">max_samples</span><span class="o">=-</span><span class="mi">1</span><span class="p">)</span>            <span class="c1"># Debe ser &gt; 0</span>
<a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a><span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">10</span><span class="p">))</span>           <span class="c1"># Dimensiones deben ser &gt; 0</span>
</code></pre></div>
<h2 id="formatos-de-datos-soportados">Formatos de Datos Soportados<a class="headerlink" href="#formatos-de-datos-soportados" title="Permanent link">&para;</a></h2>
<h3 id="estructura-de-datos-requerida">Estructura de Datos Requerida<a class="headerlink" href="#estructura-de-datos-requerida" title="Permanent link">&para;</a></h3>
<p>El DataFrame debe contener al menos estas columnas:</p>
<h4 id="columnas-obligatorias">Columnas Obligatorias<a class="headerlink" href="#columnas-obligatorias" title="Permanent link">&para;</a></h4>
<ul>
<li><code>Hugo_Symbol</code>: Símbolos de genes (ej: "TP53", "KRAS")</li>
<li><code>Variant_Classification</code>: Tipos de mutación (ej: "Missense_Mutation")</li>
<li><code>REF</code>: Alelos de referencia (ej: "A", "T")</li>
<li><code>ALT</code>: Alelos alternativos (ej: "G", "C")</li>
</ul>
<h4 id="columnas-de-muestra">Columnas de Muestra<a class="headerlink" href="#columnas-de-muestra" title="Permanent link">&para;</a></h4>
<p>Las columnas de muestra se detectan automáticamente usando patrones:
- <strong>Formato TCGA</strong>: <code>TCGA-AB-1234</code>, <code>TCGA-CD-5678</code>
- <strong>Formato GT</strong>: <code>sample1.GT</code>, <code>sample2.GT</code></p>
<h3 id="formatos-de-genotipo-soportados">Formatos de Genotipo Soportados<a class="headerlink" href="#formatos-de-genotipo-soportados" title="Permanent link">&para;</a></h3>
<p>El oncoplot soporta múltiples formatos de genotipo:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="c1"># Formatos válidos:</span>
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="s2">&quot;A|G&quot;</span>    <span class="c1"># Formato estándar pipe-separated</span>
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a><span class="s2">&quot;A/G&quot;</span>    <span class="c1"># Formato slash-separated</span>
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="s2">&quot;A:G&quot;</span>    <span class="c1"># Colon-separated</span>
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a><span class="s2">&quot;A;G&quot;</span>    <span class="c1"># Semicolon-separated</span>
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a><span class="s2">&quot;A,G&quot;</span>    <span class="c1"># Comma-separated</span>
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a><span class="c1"># Casos especiales:</span>
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a><span class="s2">&quot;A|A&quot;</span>    <span class="c1"># Homocigoto referencia (no mutación)</span>
<a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a><span class="s2">&quot;G|G&quot;</span>    <span class="c1"># Homocigoto alternativo (mutación)</span>
<a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a><span class="s2">&quot;.&quot;</span>      <span class="c1"># Datos faltantes</span>
<a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a><span class="s2">&quot;&quot;</span>       <span class="c1"># Datos vacíos</span>
</code></pre></div>
<h3 id="estructura-de-ejemplo">Estructura de Ejemplo<a class="headerlink" href="#estructura-de-ejemplo" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span>
<a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>    <span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;TP53&#39;</span><span class="p">,</span> <span class="s1">&#39;KRAS&#39;</span><span class="p">,</span> <span class="s1">&#39;PIK3CA&#39;</span><span class="p">],</span>
<a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>    <span class="s1">&#39;Variant_Classification&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Missense_Mutation&#39;</span><span class="p">,</span> <span class="s1">&#39;Nonsense_Mutation&#39;</span><span class="p">,</span> <span class="s1">&#39;In_Frame_Del&#39;</span><span class="p">],</span>
<a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a>    <span class="s1">&#39;REF&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;G&#39;</span><span class="p">],</span>
<a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a>    <span class="s1">&#39;ALT&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;G&#39;</span><span class="p">,</span> <span class="s1">&#39;T&#39;</span><span class="p">,</span> <span class="s1">&#39;A&#39;</span><span class="p">],</span>
<a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a>    <span class="s1">&#39;TCGA-AB-1234&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;A|G&#39;</span><span class="p">,</span> <span class="s1">&#39;C|C&#39;</span><span class="p">,</span> <span class="s1">&#39;G|A&#39;</span><span class="p">],</span>    <span class="c1"># Mutado en TP53 y PIK3CA</span>
<a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a>    <span class="s1">&#39;TCGA-CD-5678&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;A|A&#39;</span><span class="p">,</span> <span class="s1">&#39;C|T&#39;</span><span class="p">,</span> <span class="s1">&#39;G|G&#39;</span><span class="p">],</span>    <span class="c1"># Mutado en KRAS y PIK3CA</span>
<a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a>    <span class="s1">&#39;TCGA-EF-9012&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;A|G&#39;</span><span class="p">,</span> <span class="s1">&#39;C|T&#39;</span><span class="p">,</span> <span class="s1">&#39;G|G&#39;</span><span class="p">]</span>     <span class="c1"># Mutado en todos los genes</span>
<a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a><span class="p">})</span>
</code></pre></div>
<h2 id="ejemplos-avanzados">Ejemplos Avanzados<a class="headerlink" href="#ejemplos-avanzados" title="Permanent link">&para;</a></h2>
<h3 id="enfoque-en-genes-especificos">Enfoque en Genes Específicos<a class="headerlink" href="#enfoque-en-genes-especificos" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1"># Mostrar solo genes altamente mutados</span>
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Genes Driver Principales&quot;</span><span class="p">,</span>
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>    <span class="n">max_samples</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
<a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="mi">6</span><span class="p">)</span>
<a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="p">)</span>
</code></pre></div>
<h3 id="vista-panoramica">Vista Panorámica<a class="headerlink" href="#vista-panoramica" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="c1"># Vista amplia del paisaje mutacional</span>
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Paisaje Mutacional Completo&quot;</span><span class="p">,</span>
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>    <span class="n">max_samples</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span>
<a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a>    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">18</span><span class="p">,</span> <span class="mi">12</span><span class="p">)</span>
<a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a><span class="p">)</span>
</code></pre></div>
<h3 id="exportacion-multi-formato">Exportación Multi-formato<a class="headerlink" href="#exportacion-multi-formato" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Análisis Principal&quot;</span><span class="p">)</span>
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="c1"># Exportar en múltiples formatos</span>
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="k">for</span> <span class="n">fmt</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;png&#39;</span><span class="p">,</span> <span class="s1">&#39;pdf&#39;</span><span class="p">,</span> <span class="s1">&#39;svg&#39;</span><span class="p">]:</span>
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>    <span class="n">output_file</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;oncoplot.</span><span class="si">{</span><span class="n">fmt</span><span class="si">}</span><span class="s1">&#39;</span>
<a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a>    <span class="k">if</span> <span class="n">fmt</span> <span class="o">==</span> <span class="s1">&#39;svg&#39;</span><span class="p">:</span>
<a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a>        <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="n">output_file</span><span class="p">,</span> <span class="nb">format</span><span class="o">=</span><span class="n">fmt</span><span class="p">)</span>
<a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a>    <span class="k">else</span><span class="p">:</span>
<a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a>        <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="n">output_file</span><span class="p">,</span> <span class="nb">format</span><span class="o">=</span><span class="n">fmt</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
</code></pre></div>
<h3 id="analisis-interactivo">Análisis Interactivo<a class="headerlink" href="#analisis-interactivo" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Para exploración interactiva</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Exploración Interactiva&quot;</span><span class="p">,</span>
<a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">25</span>
<a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a><span class="p">)</span>
</code></pre></div>
<h2 id="personalizacion-visual">Personalización Visual<a class="headerlink" href="#personalizacion-visual" title="Permanent link">&para;</a></h2>
<h3 id="esquema-de-colores">Esquema de Colores<a class="headerlink" href="#esquema-de-colores" title="Permanent link">&para;</a></h3>
<p>El oncoplot usa colores estándar para tipos de mutación comunes:</p>
<table>
<thead>
<tr>
<th>Tipo de Mutación</th>
<th>Color</th>
<th>Descripción</th>
</tr>
</thead>
<tbody>
<tr>
<td>Missense_Mutation</td>
<td>Azul</td>
<td>Mutaciones que cambian aminoácido</td>
</tr>
<tr>
<td>Nonsense_Mutation</td>
<td>Rojo</td>
<td>Mutaciones que introducen stop codon</td>
</tr>
<tr>
<td>Frame_Shift_Del</td>
<td>Verde</td>
<td>Deleciones que alteran marco de lectura</td>
</tr>
<tr>
<td>Frame_Shift_Ins</td>
<td>Púrpura</td>
<td>Inserciones que alteran marco de lectura</td>
</tr>
<tr>
<td>In_Frame_Del</td>
<td>Cian claro</td>
<td>Deleciones que mantienen marco</td>
</tr>
<tr>
<td>In_Frame_Ins</td>
<td>Naranja</td>
<td>Inserciones que mantienen marco</td>
</tr>
<tr>
<td>Splice_Site</td>
<td>Rosa</td>
<td>Mutaciones en sitios de splicing</td>
</tr>
<tr>
<td>Multi_Hit</td>
<td>Negro</td>
<td>Múltiples tipos en misma muestra/gen</td>
</tr>
<tr>
<td>None</td>
<td>Gris claro</td>
<td>Sin mutación detectada</td>
</tr>
</tbody>
</table>
<h3 id="tamanos-de-figura-recomendados">Tamaños de Figura Recomendados<a class="headerlink" href="#tamanos-de-figura-recomendados" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1"># Para pocos genes, muchas muestras</span>
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="mi">6</span><span class="p">)</span>   <span class="c1"># Formato panorámico</span>
<a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>
<a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a><span class="c1"># Para muchos genes, pocas muestras  </span>
<a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">16</span><span class="p">)</span>  <span class="c1"># Formato vertical</span>
<a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a>
<a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a><span class="c1"># Balanceado general</span>
<a id="__codelineno-11-8" name="__codelineno-11-8" href="#__codelineno-11-8"></a><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span> <span class="mi">10</span><span class="p">)</span>  <span class="c1"># Formato estándar</span>
<a id="__codelineno-11-9" name="__codelineno-11-9" href="#__codelineno-11-9"></a>
<a id="__codelineno-11-10" name="__codelineno-11-10" href="#__codelineno-11-10"></a><span class="c1"># Para presentaciones</span>
<a id="__codelineno-11-11" name="__codelineno-11-11" href="#__codelineno-11-11"></a><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">24</span><span class="p">,</span> <span class="mi">12</span><span class="p">)</span>  <span class="c1"># Extra grande</span>
</code></pre></div>
<h2 id="interpretacion-del-oncoplot">Interpretación del Oncoplot<a class="headerlink" href="#interpretacion-del-oncoplot" title="Permanent link">&para;</a></h2>
<h3 id="lectura-del-grafico">Lectura del Gráfico<a class="headerlink" href="#lectura-del-grafico" title="Permanent link">&para;</a></h3>
<ol>
<li><strong>Eje Y (Genes)</strong>: Ordenados por frecuencia de mutación (más mutados arriba)</li>
<li><strong>Eje X (Muestras)</strong>: Ordenadas por carga mutacional total</li>
<li><strong>Colores</strong>: Indican tipo de mutación según leyenda</li>
<li><strong>Celdas grises</strong>: No hay mutación detectada</li>
<li><strong>Celdas negras</strong>: Múltiples tipos de mutación (Multi_Hit)</li>
</ol>
<h3 id="patrones-importantes">Patrones Importantes<a class="headerlink" href="#patrones-importantes" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Genes frecuentemente mutados</strong>: Aparecen en la parte superior</li>
<li><strong>Muestras con alta carga mutacional</strong>: Hacia la izquierda</li>
<li><strong>Co-ocurrencias</strong>: Genes mutados en las mismas muestras</li>
<li><strong>Exclusividad mutua</strong>: Genes que raramente se mutan juntos</li>
</ul>
<h3 id="analisis-de-multi_hit">Análisis de Multi_Hit<a class="headerlink" href="#analisis-de-multi_hit" title="Permanent link">&para;</a></h3>
<p>Cuando una muestra tiene múltiples tipos de mutación en el mismo gen:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="c1"># Ejemplo: TP53 con Missense y Nonsense en la misma muestra</span>
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a><span class="c1"># Se mostrará como Multi_Hit (negro)</span>
<a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a><span class="n">data_multihit</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span>
<a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a>    <span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;TP53&#39;</span><span class="p">,</span> <span class="s1">&#39;TP53&#39;</span><span class="p">],</span>
<a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a>    <span class="s1">&#39;Variant_Classification&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Missense_Mutation&#39;</span><span class="p">,</span> <span class="s1">&#39;Nonsense_Mutation&#39;</span><span class="p">],</span>
<a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a>    <span class="s1">&#39;REF&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">],</span>
<a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a>    <span class="s1">&#39;ALT&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;G&#39;</span><span class="p">,</span> <span class="s1">&#39;T&#39;</span><span class="p">],</span>
<a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a>    <span class="s1">&#39;TCGA-AB-1234&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;A|G&#39;</span><span class="p">,</span> <span class="s1">&#39;C|T&#39;</span><span class="p">]</span>  <span class="c1"># Ambas mutaciones presentes</span>
<a id="__codelineno-12-9" name="__codelineno-12-9" href="#__codelineno-12-9"></a><span class="p">})</span>
</code></pre></div>
<h2 id="solucion-de-problemas">Solución de Problemas<a class="headerlink" href="#solucion-de-problemas" title="Permanent link">&para;</a></h2>
<h3 id="errores-comunes">Errores Comunes<a class="headerlink" href="#errores-comunes" title="Permanent link">&para;</a></h3>
<h4 id="1-columnas-faltantes">1. Columnas Faltantes<a class="headerlink" href="#1-columnas-faltantes" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="c1"># Error: ValueError: Faltan las siguientes columnas requeridas: [&#39;REF&#39;]</span>
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="c1"># Solución: Verificar que todas las columnas requeridas estén presentes</span>
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="n">required_columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">,</span> <span class="s1">&#39;Variant_Classification&#39;</span><span class="p">,</span> <span class="s1">&#39;REF&#39;</span><span class="p">,</span> <span class="s1">&#39;ALT&#39;</span><span class="p">]</span>
<a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a><span class="n">missing</span> <span class="o">=</span> <span class="p">[</span><span class="n">col</span> <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">required_columns</span> <span class="k">if</span> <span class="n">col</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">data</span><span class="o">.</span><span class="n">columns</span><span class="p">]</span>
<a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Columnas faltantes: </span><span class="si">{</span><span class="n">missing</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</code></pre></div>
<h4 id="2-no-se-detectan-muestras">2. No se Detectan Muestras<a class="headerlink" href="#2-no-se-detectan-muestras" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="c1"># Error: No se pudieron detectar columnas de muestra automáticamente</span>
<a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="c1"># Solución: Verificar formato de nombres de columnas</span>
<a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="n">sample_cols</span> <span class="o">=</span> <span class="p">[</span><span class="n">col</span> <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">data</span><span class="o">.</span><span class="n">columns</span> <span class="k">if</span> <span class="n">col</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;TCGA-&#39;</span><span class="p">)]</span>
<a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Columnas TCGA encontradas: </span><span class="si">{</span><span class="n">sample_cols</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</code></pre></div>
<h4 id="3-no-hay-datos-de-mutacion">3. No Hay Datos de Mutación<a class="headerlink" href="#3-no-hay-datos-de-mutacion" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="c1"># Error: No hay datos de mutación para visualizar</span>
<a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="c1"># Solución: Verificar que hay mutaciones detectables</span>
<a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a><span class="n">mutations_detected</span> <span class="o">=</span> <span class="mi">0</span>
<a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">sample_cols</span><span class="p">[:</span><span class="mi">5</span><span class="p">]:</span>  <span class="c1"># Revisar primeras 5 muestras</span>
<a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a>    <span class="n">mutations</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;|&#39;</span> <span class="ow">in</span> <span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="ow">and</span> <span class="s1">&#39;A|A&#39;</span> <span class="ow">not</span> <span class="ow">in</span> <span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a>    <span class="n">mutations_detected</span> <span class="o">+=</span> <span class="n">mutations</span>
<a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Mutaciones detectadas en primeras 5 muestras: </span><span class="si">{</span><span class="n">mutations_detected</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</code></pre></div>
<h4 id="4-parametros-invalidos">4. Parámetros Inválidos<a class="headerlink" href="#4-parametros-invalidos" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="c1"># Error: ValueError: top_genes_count debe ser un entero positivo</span>
<a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a><span class="c1"># Solución: Usar valores válidos</span>
<a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a><span class="k">assert</span> <span class="n">top_genes_count</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">,</span> <span class="s2">&quot;top_genes_count debe ser &gt; 0&quot;</span>
<a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="k">assert</span> <span class="n">max_samples</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">,</span> <span class="s2">&quot;max_samples debe ser &gt; 0&quot;</span>
<a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a><span class="k">assert</span> <span class="nb">len</span><span class="p">(</span><span class="n">figsize</span><span class="p">)</span> <span class="o">==</span> <span class="mi">2</span> <span class="ow">and</span> <span class="nb">all</span><span class="p">(</span><span class="n">x</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">figsize</span><span class="p">),</span> <span class="s2">&quot;figsize debe ser (ancho, alto) con valores &gt; 0&quot;</span>
</code></pre></div>
<h3 id="debugging-paso-a-paso">Debugging Paso a Paso<a class="headerlink" href="#debugging-paso-a-paso" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="c1"># 1. Verificar estructura de datos</span>
<a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Columnas disponibles:&quot;</span><span class="p">,</span> <span class="n">data</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
<a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Forma de datos:&quot;</span><span class="p">,</span> <span class="n">data</span><span class="o">.</span><span class="n">shape</span><span class="p">)</span>
<a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a>
<a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a><span class="c1"># 2. Verificar detección de muestras</span>
<a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pyMut.visualizations.oncoplot</span><span class="w"> </span><span class="kn">import</span> <span class="n">detect_sample_columns</span>
<a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a><span class="k">try</span><span class="p">:</span>
<a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a>    <span class="n">samples</span> <span class="o">=</span> <span class="n">detect_sample_columns</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-17-9" name="__codelineno-17-9" href="#__codelineno-17-9"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Muestras detectadas: </span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">samples</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-17-10" name="__codelineno-17-10" href="#__codelineno-17-10"></a><span class="k">except</span> <span class="ne">ValueError</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
<a id="__codelineno-17-11" name="__codelineno-17-11" href="#__codelineno-17-11"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Error en detección: </span><span class="si">{</span><span class="n">e</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-17-12" name="__codelineno-17-12" href="#__codelineno-17-12"></a>
<a id="__codelineno-17-13" name="__codelineno-17-13" href="#__codelineno-17-13"></a><span class="c1"># 3. Verificar datos de mutación</span>
<a id="__codelineno-17-14" name="__codelineno-17-14" href="#__codelineno-17-14"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pyMut.visualizations.oncoplot</span><span class="w"> </span><span class="kn">import</span> <span class="n">process_mutation_matrix</span>
<a id="__codelineno-17-15" name="__codelineno-17-15" href="#__codelineno-17-15"></a><span class="k">try</span><span class="p">:</span>
<a id="__codelineno-17-16" name="__codelineno-17-16" href="#__codelineno-17-16"></a>    <span class="n">matrix</span><span class="p">,</span> <span class="n">counts</span> <span class="o">=</span> <span class="n">process_mutation_matrix</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-17-17" name="__codelineno-17-17" href="#__codelineno-17-17"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Matriz generada: </span><span class="si">{</span><span class="n">matrix</span><span class="o">.</span><span class="n">shape</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-17-18" name="__codelineno-17-18" href="#__codelineno-17-18"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Genes con mutaciones: </span><span class="si">{</span><span class="nb">len</span><span class="p">([</span><span class="n">g</span><span class="w"> </span><span class="k">for</span><span class="w"> </span><span class="n">g</span><span class="p">,</span><span class="w"> </span><span class="n">c</span><span class="w"> </span><span class="ow">in</span><span class="w"> </span><span class="n">counts</span><span class="o">.</span><span class="n">items</span><span class="p">()</span><span class="w"> </span><span class="k">if</span><span class="w"> </span><span class="n">c</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="mi">0</span><span class="p">])</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-17-19" name="__codelineno-17-19" href="#__codelineno-17-19"></a><span class="k">except</span> <span class="ne">Exception</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
<a id="__codelineno-17-20" name="__codelineno-17-20" href="#__codelineno-17-20"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Error en procesamiento: </span><span class="si">{</span><span class="n">e</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<a id="__codelineno-17-21" name="__codelineno-17-21" href="#__codelineno-17-21"></a>
<a id="__codelineno-17-22" name="__codelineno-17-22" href="#__codelineno-17-22"></a><span class="c1"># 4. Generar oncoplot</span>
<a id="__codelineno-17-23" name="__codelineno-17-23" href="#__codelineno-17-23"></a><span class="k">try</span><span class="p">:</span>
<a id="__codelineno-17-24" name="__codelineno-17-24" href="#__codelineno-17-24"></a>    <span class="n">py_mut</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<a id="__codelineno-17-25" name="__codelineno-17-25" href="#__codelineno-17-25"></a>    <span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span><span class="n">top_genes_count</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
<a id="__codelineno-17-26" name="__codelineno-17-26" href="#__codelineno-17-26"></a>    <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;✓ Oncoplot generado exitosamente&quot;</span><span class="p">)</span>
<a id="__codelineno-17-27" name="__codelineno-17-27" href="#__codelineno-17-27"></a><span class="k">except</span> <span class="ne">Exception</span> <span class="k">as</span> <span class="n">e</span><span class="p">:</span>
<a id="__codelineno-17-28" name="__codelineno-17-28" href="#__codelineno-17-28"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Error en oncoplot: </span><span class="si">{</span><span class="n">e</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</code></pre></div>
<h3 id="problemas-de-rendimiento">Problemas de Rendimiento<a class="headerlink" href="#problemas-de-rendimiento" title="Permanent link">&para;</a></h3>
<h4 id="datasets-grandes">Datasets Grandes<a class="headerlink" href="#datasets-grandes" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="c1"># Para datasets con &gt;1000 genes o &gt;500 muestras</span>
<a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-18-3" name="__codelineno-18-3" href="#__codelineno-18-3"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>     <span class="c1"># Limitar genes</span>
<a id="__codelineno-18-4" name="__codelineno-18-4" href="#__codelineno-18-4"></a>    <span class="n">max_samples</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span>        <span class="c1"># Limitar muestras  </span>
<a id="__codelineno-18-5" name="__codelineno-18-5" href="#__codelineno-18-5"></a>    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span> <span class="mi">8</span><span class="p">)</span>         <span class="c1"># Tamaño moderado</span>
<a id="__codelineno-18-6" name="__codelineno-18-6" href="#__codelineno-18-6"></a><span class="p">)</span>
</code></pre></div>
<h4 id="memoria-insuficiente">Memoria Insuficiente<a class="headerlink" href="#memoria-insuficiente" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="c1"># Procesar en chunks para datasets muy grandes</span>
<a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="n">chunk_size</span> <span class="o">=</span> <span class="mi">50</span>
<a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a><span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="p">),</span> <span class="n">chunk_size</span><span class="p">):</span>
<a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a>    <span class="n">chunk</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">:</span><span class="n">i</span><span class="o">+</span><span class="n">chunk_size</span><span class="p">]</span>
<a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a>    <span class="n">py_mut_chunk</span> <span class="o">=</span> <span class="n">PyMutation</span><span class="p">(</span><span class="n">chunk</span><span class="p">)</span>
<a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a>    <span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut_chunk</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-19-7" name="__codelineno-19-7" href="#__codelineno-19-7"></a>        <span class="n">title</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;Chunk </span><span class="si">{</span><span class="n">i</span><span class="o">//</span><span class="n">chunk_size</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="mi">1</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
<a id="__codelineno-19-8" name="__codelineno-19-8" href="#__codelineno-19-8"></a>        <span class="n">max_samples</span><span class="o">=</span><span class="n">chunk_size</span>
<a id="__codelineno-19-9" name="__codelineno-19-9" href="#__codelineno-19-9"></a>    <span class="p">)</span>
<a id="__codelineno-19-10" name="__codelineno-19-10" href="#__codelineno-19-10"></a>    <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;oncoplot_chunk_</span><span class="si">{</span><span class="n">i</span><span class="o">//</span><span class="n">chunk_size</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="mi">1</span><span class="si">}</span><span class="s1">.png&#39;</span><span class="p">)</span>
</code></pre></div>
<h2 id="optimizacion-de-rendimiento">Optimización de Rendimiento<a class="headerlink" href="#optimizacion-de-rendimiento" title="Permanent link">&para;</a></h2>
<h3 id="mejores-practicas">Mejores Prácticas<a class="headerlink" href="#mejores-practicas" title="Permanent link">&para;</a></h3>
<ol>
<li><strong>Filtrar datos antes</strong>: Remover genes/muestras irrelevantes</li>
<li><strong>Usar parámetros apropiados</strong>: Limitar <code>top_genes_count</code> y <code>max_samples</code></li>
<li><strong>Configurar calidad una vez</strong>: Llamar <code>configure_high_quality_plots()</code> al inicio</li>
<li><strong>Cerrar figuras</strong>: Usar <code>plt.close(fig)</code> para liberar memoria</li>
</ol>
<h3 id="parametros-de-rendimiento">Parámetros de Rendimiento<a class="headerlink" href="#parametros-de-rendimiento" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="c1"># Para análisis exploratorio rápido</span>
<a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span>
<a id="__codelineno-20-4" name="__codelineno-20-4" href="#__codelineno-20-4"></a>    <span class="n">max_samples</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span>
<a id="__codelineno-20-5" name="__codelineno-20-5" href="#__codelineno-20-5"></a>    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">)</span>
<a id="__codelineno-20-6" name="__codelineno-20-6" href="#__codelineno-20-6"></a><span class="p">)</span>
<a id="__codelineno-20-7" name="__codelineno-20-7" href="#__codelineno-20-7"></a>
<a id="__codelineno-20-8" name="__codelineno-20-8" href="#__codelineno-20-8"></a><span class="c1"># Para análisis final de alta calidad</span>
<a id="__codelineno-20-9" name="__codelineno-20-9" href="#__codelineno-20-9"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">(</span>
<a id="__codelineno-20-10" name="__codelineno-20-10" href="#__codelineno-20-10"></a>    <span class="n">top_genes_count</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
<a id="__codelineno-20-11" name="__codelineno-20-11" href="#__codelineno-20-11"></a>    <span class="n">max_samples</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
<a id="__codelineno-20-12" name="__codelineno-20-12" href="#__codelineno-20-12"></a>    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="mi">12</span><span class="p">)</span>
<a id="__codelineno-20-13" name="__codelineno-20-13" href="#__codelineno-20-13"></a><span class="p">)</span>
</code></pre></div>
<h3 id="monitoreo-de-recursos">Monitoreo de Recursos<a class="headerlink" href="#monitoreo-de-recursos" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="kn">import</span><span class="w"> </span><span class="nn">time</span>
<a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="kn">import</span><span class="w"> </span><span class="nn">psutil</span>
<a id="__codelineno-21-3" name="__codelineno-21-3" href="#__codelineno-21-3"></a>
<a id="__codelineno-21-4" name="__codelineno-21-4" href="#__codelineno-21-4"></a><span class="c1"># Monitorear tiempo y memoria</span>
<a id="__codelineno-21-5" name="__codelineno-21-5" href="#__codelineno-21-5"></a><span class="n">start_time</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<a id="__codelineno-21-6" name="__codelineno-21-6" href="#__codelineno-21-6"></a><span class="n">memory_before</span> <span class="o">=</span> <span class="n">psutil</span><span class="o">.</span><span class="n">Process</span><span class="p">()</span><span class="o">.</span><span class="n">memory_info</span><span class="p">()</span><span class="o">.</span><span class="n">rss</span> <span class="o">/</span> <span class="mi">1024</span> <span class="o">/</span> <span class="mi">1024</span>  <span class="c1"># MB</span>
<a id="__codelineno-21-7" name="__codelineno-21-7" href="#__codelineno-21-7"></a>
<a id="__codelineno-21-8" name="__codelineno-21-8" href="#__codelineno-21-8"></a><span class="n">fig</span> <span class="o">=</span> <span class="n">py_mut</span><span class="o">.</span><span class="n">oncoplot</span><span class="p">()</span>
<a id="__codelineno-21-9" name="__codelineno-21-9" href="#__codelineno-21-9"></a>
<a id="__codelineno-21-10" name="__codelineno-21-10" href="#__codelineno-21-10"></a><span class="n">end_time</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<a id="__codelineno-21-11" name="__codelineno-21-11" href="#__codelineno-21-11"></a><span class="n">memory_after</span> <span class="o">=</span> <span class="n">psutil</span><span class="o">.</span><span class="n">Process</span><span class="p">()</span><span class="o">.</span><span class="n">memory_info</span><span class="p">()</span><span class="o">.</span><span class="n">rss</span> <span class="o">/</span> <span class="mi">1024</span> <span class="o">/</span> <span class="mi">1024</span>  <span class="c1"># MB</span>
<a id="__codelineno-21-12" name="__codelineno-21-12" href="#__codelineno-21-12"></a>
<a id="__codelineno-21-13" name="__codelineno-21-13" href="#__codelineno-21-13"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Tiempo: </span><span class="si">{</span><span class="n">end_time</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">start_time</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> segundos&quot;</span><span class="p">)</span>
<a id="__codelineno-21-14" name="__codelineno-21-14" href="#__codelineno-21-14"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Memoria usada: </span><span class="si">{</span><span class="n">memory_after</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">memory_before</span><span class="si">:</span><span class="s2">.1f</span><span class="si">}</span><span class="s2"> MB&quot;</span><span class="p">)</span>
</code></pre></div>
<hr />
<h2 id="mas-recursos">Más Recursos<a class="headerlink" href="#mas-recursos" title="Permanent link">&para;</a></h2>
<ul>
<li><strong>Guía de Instalación</strong>: <a href="../">docs/index.md</a></li>
<li><strong>Código Fuente</strong>: <a href="../src/pyMut/visualizations/oncoplot.py">src/pyMut/visualizations/oncoplot.py</a></li>
<li><strong>Ejemplos Completos</strong>: <a href="../examples/example_oncoplot.py">examples/example_oncoplot.py</a></li>
<li><strong>Tests</strong>: <a href="../tests/test_oncoplot.py">tests/test_oncoplot.py</a></li>
</ul>
<p>Para preguntas específicas o reportar problemas, consulta la documentación del proyecto o contacta al equipo de desarrollo. </p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 pyMut Development Team
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.code.copy", "navigation.instant", "navigation.tracking", "search.highlight", "search.suggest"], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.50899def.min.js"></script>
      
    
  </body>
</html>